---
title: "TRS-BIOME Metagenomics OSCA analysis prep"
author: "by Svetlina Vasileva - `r format(Sys.time(), '%d %B %Y')`"
output:
  epuRate::UQ:
    toc: TRUE
    code_folding: "hide"
  html_document:
    self_contained: no
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) #silences messages and warnings to make the script cleaner
```

# Metagenomics OSCA analysis overview

**Motivation**

- calculate variance explained by bacterial microbiome and functional data
- look for associations between phenotypes (SCZ, treatment-response, clozapine use, dietary PCs, Bristol Stool Scale) and the microbiome in a mixed linear model framework
- dissect relative contributions of taxonomic and functional information to SCZ, dietary PCs and Bristol Stool Chart (--mgrm analyses)

**Datasets**

- phenotypes: 
    - NC/SCZ
    - APP/CLZ
    - CR/CNR
    - 4 dietary PCs
    - Bristol Stool Scale
- OSCA matrices:
    - taxonomic 
        - collapsed to species, genera, family levels
    - functional
        - Enzyme Commission (collapsed to level 4 and level 3)
        - TCDB (focused on transporters)
        - MetaCyc pathways

**Methods**

- calculate the median read count for each taxa/functional variable (different taxonomic levels shown below: species, genera, family)
    - "common" taxa/functional variables: filter out taxa with median read count < 0 (motivation: reduce zero-inflated taxa)
    - "rare" taxa/functional variables: filter out variable with <10 non-zero values + filter out variable with median read count > 0
- perform centred-log-ratio transformation, and check distribution to have centrality (suggested by Restu)
- create matrix for use to input into OSCA
- OSCA analyses (see script: metagenomics_analysis_osca.sh)
    - variance explained

# Pre-OSCA prep

- (copied over from metagenomics_familytaxa_matrix.R)

## Libraries and directories

```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(rmarkdown)    # install.packages("rmarkdown") 
library(epuRate)      # devtools::install_github("holtzy/epuRate", force=TRUE)
library(DT)
library(mixOmics)
library(hrbrthemes)
library(viridis)
#hrbrthemes::import_roboto_condensed()

library(data.table)
library(stringr)
library(tidyverse)
library(compositions)
options(stringsAsFactors = FALSE)
library(ggrepel)
library(reshape2)
library(qqman)
library(R.utils) # to open .gz files
library(RNOmni) # for the RankNorm function to the inverse rank transformation
```

```{r}
rm(list = ls())
setwd("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data")
windowsFonts(Times=windowsFont("TT Times New Roman"))

# Directories
taxa_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/taxonomic_profiles_relabu/profiles.tsv"
taxa_count_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/taxonomic_profiles_counts/profiles.tsv"
ec_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/functional_profiles/by_sample/EC.samples.tsv.gz"
tcdb_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/functional_profiles/by_sample/TCDB.samples.tsv.gz"
mcpath_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/functional_profiles/by_sample/MetaCyc_pathway.samples.tsv.gz"
func_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/functional_profiles/by_sample/MICROBA.samples.tsv.gz"
clinical_full_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/trs_biome_full_clinical_ipaq_sagis_scfa_dates_arfs_pe13_pcs_int.csv" # Includes SCFA, dates, IPAQ, SAGIS, arfs, 13 clr PCS from % energy (pe) from 13 food groups from AES + BMI and 4PC INT     
sample_qc_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/Metagenomics/processed_data/sample_qc.tsv" # 
ap_meds_dir <- ("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/trs_biome_aap_bnf_amh.csv")


# taxa_count_dir <- "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/processed_data/taxonomic_profiles/profiles_counts_MGDBv2ext.tsv" # not sure what this file is

# Colour schemes for plots
color_all<-c("#D81B60","#21DAC9", "#FFC107", "#3A9A8A")
color_nc_scz<-c("#D81B60", "#E8913C")
color_aap_clz<-c("#102C8E", "#21DAC9")
color_cr_cnr<-c("#FFC107", "#3A9A8A")

# Load data
clinical_full <- read.csv(clinical_full_dir, header = T, as.is = T)
ap_meds <- read.csv(ap_meds_dir, header = T, as.is = T) # AP meds doses and AMH/BNF

```

## Data cleaning

### Clinical data cleaning

```{r}

 
clinical_full$participant_group <- factor(clinical_full$participant_group, levels=c('NC', 'AAP', 'CR', 'CNR'))
clinical_full$panss_total <- as.numeric(clinical_full$panss_total)

# Add dose factors: AMH (Australian) and BNF (British)
clinical_full<-clinical_full %>% 
  right_join(ap_meds[,c("participant_id", "CLZ_factor_AMH", "Total_BNF","Total_AMH", "cloz_dose", "cloz_cd_ratio")], by="participant_id") # dose factors for CLZ are the same with according to both BNF and AMH

# Get years 
clinical_full$duration_illness[clinical_full$duration_illness=="9 months"]<-0.75 # turn to years
clinical_full$duration_illness<-as.numeric(clinical_full$duration_illness)
clinical_full$duration_first_treatment[clinical_full$duration_first_treatment=="9 months"]<-0.75 # turn to years
clinical_full$duration_first_treatment<-as.numeric(clinical_full$duration_first_treatment)


# Extract selected columns from clinical_full file
select_col <- c("MG_SampleID", "participant_id", "participant_group", "age", "sex", "bmi", "bmi_int", "scz", "clz","stool_type_regroup","PC1", "PC2", "PC3", "PC4","PC5", "PC6", "PC7", "PC8", "PC9", "PC10","PC11", "PC12", "PC13","PC1_int", "PC2_int", "PC3_int", "PC4_int" , "living_situation", "smoking_status", "audit_total", "panss_positive_total", "panss_negative_total", "panss_general_total", "panss_total", "total_depression", "total_anxiety", "total_stress", "sofas_score", "sagis_total_burden", "ipaq_met", "ipaq_pacat", "hba1c", "fasting_glucose", "fasting_tg", "fasting_hdl", "fasting_ldl", "total_cholesterol", "cloz_level", "norcloz_level", "cloz_norcloz_ratio", "arfs", "TAP", "AAP", "metformin", "laxatives", "PPI", "SSRI", "SNIR", "other_antidepressants", "acetate_clean", "butyrate_clean", "propionate_clean", "mets_5_notes", "mets_2","sagis_10","sagis_constipation_only", "Total_BNF", "Total_AMH", "CLZ_factor_AMH", "cloz_dose", "cloz_cd_ratio", "duration_first_treatment", "duration_illness")
# create a vector with relevant column names
diet<-clinical_full %>% dplyr::select(c(select_col))    # take these columns

# Sort the pheno/covariates file
diet <- diet[order(diet$MG_SampleID),]


```

#### Plot distributions

```{r}

# Check distribution of AMH/BNF factors and normalise if necessary

ggplot((diet %>% filter(participant_group != "NC")), aes(x=Total_AMH)) +
  geom_density() +
  scale_x_continuous(limits = c(min(diet$Total_AMH), max(diet$Total_AMH))) +
  labs(title = "Density Plot for Total_AMH")

ggplot((diet), aes(x=arfs)) +
  geom_density() 

ggplot((diet %>% filter(participant_group != "NC")), aes(x=Total_BNF)) +
  geom_density() +
  labs(title = "Density Plot for Total_BNF")

ggplot((diet %>% filter(participant_group %in% c("CR","CNR"))), aes(x=CLZ_factor_AMH)) +
  geom_density() +
  labs(title = "Density Plot for CLZ_factor_AMH")

ggplot((diet %>% filter(participant_group != "NC")), aes(x=panss_total)) +
  geom_density() +
  labs(title = "Density Plot for panss_total")


# Cloz dose x cloz blood levels
ggplot(diet, aes(x=CLZ_factor_AMH, y=cloz_level)) +
  geom_point()

```

#### Transform to normalise (inverse rank tranformation, INT)

```{r}

# Normalise variables  - have to remove NAs before transformation and omit.na doesn't work for some reason -> subsample only participants that have those measures before transform
diet <- diet %>% 
  mutate(acetate_clean_int = RankNorm(acetate_clean),
         butyrate_clean_int = RankNorm(butyrate_clean),
         propionate_clean_int = RankNorm(propionate_clean),
         arfs_int = RankNorm(arfs))
diet_scz_only <- diet %>% 
        filter(participant_group != "NC") %>% 
        mutate(panss_total_int = RankNorm(panss_total),
         Total_BNF_int = RankNorm(Total_BNF),
         Total_AMH_int = RankNorm(Total_AMH))
diet_clz_only <- diet %>% 
  filter(participant_group  %in% c("CR","CNR")) %>% 
  mutate(CLZ_factor_AMH_int = RankNorm(CLZ_factor_AMH),
         cloz_dose_int = RankNorm(cloz_dose))
diet_clz_46 <- diet %>% 
  filter(cloz_level  != "NA") %>%
  mutate(cloz_level_int = RankNorm(cloz_level),
         norcloz_level_int = RankNorm(norcloz_level),
         cloz_norcloz_ratio_int = RankNorm(cloz_norcloz_ratio), 
         cloz_cd_ratio_int = RankNorm(cloz_cd_ratio),)

# Add the transfromed variables to the main dataframe
diet <- diet %>%
  left_join(diet_scz_only[,c("MG_SampleID", "panss_total_int", "Total_BNF_int", "Total_AMH_int")], by = "MG_SampleID") %>%  
  left_join(diet_clz_only[,c("MG_SampleID", "CLZ_factor_AMH_int", "cloz_dose_int")], by = "MG_SampleID") %>%
  left_join(diet_clz_46[,c("MG_SampleID", "cloz_level_int", "norcloz_level_int", "cloz_norcloz_ratio_int", "cloz_cd_ratio_int")], by = "MG_SampleID")

```

### Taxa count tables

```{r, echo=FALSE, message=FALSE, warning=FALSE}

# Read in data
taxa <- fread(taxa_dir)
taxa_count <- fread(taxa_count_dir)
sample_qc <- read.delim(sample_qc_dir, header = T, as.is = T)

# Rename columns
count_cols <- grep("BBK", colnames(taxa_count))
colnames(taxa_count)[count_cols] <- substring(colnames(taxa_count)[count_cols], 1, 7)
colnames(taxa_count)[2] <- "GTDB.taxonomy"

# Take individuals included in this analysis
taxa_count <- data.frame(taxa_count)
taxa_count_keep <- which(colnames(taxa_count) %in% diet$MG_SampleID)
taxa_count <- data.frame(Taxon = taxa_count$Taxon, 
	GTDB.taxonomy = taxa_count$GTDB.taxonomy, 
	sapply(taxa_count[,taxa_count_keep], function(x) x/2), stringsAsFactors = FALSE) # divide read counts by 2 in updated MCPv2

# Add column of unassigned reads in count table
count_cols <- grep("BBK", colnames(taxa_count))

colnames(sample_qc)[which(colnames(sample_qc) == "ID")] <- "SampleID"
sample_qc$SampleID <- substring(sample_qc$SampleID, 1, 7)
sample_qc <- sample_qc %>% filter(SampleID %in% diet$MG_SampleID) %>% arrange(SampleID)
sum.tmp <- colSums(data.frame(taxa_count[,count_cols]))

identical(as.character(names(sum.tmp)), colnames(taxa_count[,count_cols]))

unassigned <- as.numeric(sample_qc$TOTAL_DOWNSAMPLED - sum.tmp)
names(unassigned) <- colnames(taxa_count)[3:ncol(taxa_count)]
# taxa_count <- rbind(taxa_count, c("UNASSIGNED", "UNASSIGNED", unassigned))
taxa_count[nrow(taxa_count) + 1,] <- NA
taxa_count[nrow(taxa_count), "Taxon"] <- "UNASSIGNED"
taxa_count[nrow(taxa_count), "GTDB.taxonomy"] <- "UNASSIGNED"
taxa_count[nrow(taxa_count), 3:ncol(taxa_count)] <- unassigned

taxa_count_keep <- taxa_count # Do this so can re-use code when subsetting by genus

```

#### Density plots for read counts per group
- CR have lower levels of total reads assigned
- multi-density plot [https://r-graph-gallery.com/135-stacked-density-graph.html]

```{r}
# Creating the files

total_count<-colSums(taxa_count[,3:ncol(taxa_count)])
total_count<-as.matrix(total_count)
colnames(total_count)<-c("total_count")

unassigned_count<-taxa_count[c(1432),3:ncol(taxa_count)]
unassigned_count<-as.matrix(unassigned_count)
unassigned_count<-t(unassigned_count)
colnames(unassigned_count)<-c("unassigned_count")

plot_counts<-cbind(total_count,unassigned_count)
plot_counts<-data.frame(plot_counts)
plot_counts$total_assigned<-plot_counts$total_count-plot_counts$unassigned_count
groups<-diet[,c("MG_SampleID", "participant_group", "clz", "scz")]
plot_counts$MG_SampleID<-rownames(plot_counts) # turn rownames into a column
plot_counts<-groups %>% dplyr::left_join (plot_counts, by="MG_SampleID")
plot_counts$participant_group<-factor(plot_counts$participant_group, levels=c('NC', 'AAP', 'CR', 'CNR')) 

ggplot(data=plot_counts, aes(x=unassigned_count)) +
  geom_density()
```


```{r}
# Ploting the distribution of reads

# Unassigned_count - density plot 4 groups
ggplot(data=plot_counts, aes(x=unassigned_count, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4)  +
    scale_fill_manual(values=c("#D81B60","#21DAC9", "#FFC107", "#3A9A8A")) +
  ylim(0,2.1e-06)

# Total_assigned - density plot - 4 groups
ggplot(data=plot_counts, aes(x=total_assigned, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4)  +
    scale_fill_manual(values=c("#D81B60","#21DAC9", "#FFC107", "#3A9A8A")) +
  ylim(0,2.1e-06)

# Total_assigned - density plot - NC vs SCZ
ggplot(data=plot_counts, aes(x=total_assigned, group=scz, fill=scz)) +
    geom_density(adjust=1.5, alpha=.4) +
    scale_fill_manual(values=c("#D81B60", "#E8913C"), name = "NC and SCZ", labels = c("NC", "SCZ")) +
  ylim(0,2.1e-06) 

# Total_assigned - density plot - AAP vs CLZ
ggplot(data=subset(plot_counts, scz %in% "scz"), aes(x=total_assigned, group=clz, fill=clz)) +
    geom_density(adjust=1.5, alpha=.4) +
    scale_fill_manual(values=c("#21DAC9", "#102C8E"), name = "People with SCZ", labels = c("AAP", "CLZ")) +
  ylim(0,2.1e-06)

# Total_assigned - density plot - CR vs CNR
ggplot(data=subset(plot_counts, clz %in% "clz"), aes(x=total_assigned, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4) +
    scale_fill_manual(values=c("#FFC107", "#3A9A8A"), name = "CLZ users", labels = c("CR", "CNR")) +
  ylim(0,2.1e-06)

 
```


### Functional tables

#### Enzyme Commission

```{r}
ec <- fread(ec_dir)

# Take individuals included in this analysis
ec <- data.frame(ec)
ec_keep <- which(colnames(ec) %in% diet$MG_SampleID)

# Left out the UNASSIGNED column generation
ec_keep <- ec

```

#### TCDB

- Left out the UNASSIGNED column generation, as it looks like the numbers don't quite add up ...

```{r}
tcdb <- fread(tcdb_dir)

# Take individuals included in this analysis
tcdb <- data.frame(tcdb)
tcdb_keep <- which(colnames(tcdb) %in% diet$MG_SampleID)

# Left out the UNASSIGNED column generation
tcdb_keep <- tcdb

```

#### MetaCyc Pathways

```{r}
mcpath <- fread(mcpath_dir)

# Take individuals included in this analysis
 
mcpath_keep <- which(colnames(mcpath) %in% diet$MG_SampleID)

# Left out the UNASSIGNED column generation
mcpath_keep <- mcpath

```

#### Microba functional table

```{r}
func <- fread(func_dir)

# Take individuals included in this analysis
func <- data.frame(func)
func_keep <- which(colnames(func) %in% diet$MG_SampleID)

# Left out the UNASSIGNED column generation
func_keep <- func

```

## Collapse to species

### Take species OTUs

```{r}

taxa_count <- taxa_count_keep

species <- taxa_count$Taxon
species <- substring(species, 4) # removes "s__" 
taxa_count$Taxon <- species

colnames(taxa_count)[which(colnames(taxa_count) == "Taxon")] <- "Species"
taxa_count[which(taxa_count$GTDB.taxonomy == "UNASSIGNED"), "Species"] <- "UNASSIGNED"

```

### Collapse

```{r}

species_uniq <- unique(species)
count_cols <- grep("BBK", colnames(taxa_count))

taxa_count_species.mat <- matrix(nrow = length(species_uniq), ncol = length(count_cols))
for (i in 1:length(species_uniq)) {
	taxa_count_species.mat[i,] <- colSums(taxa_count[which(taxa_count$Species == species_uniq[i]),count_cols])
}
colnames(taxa_count_species.mat) <- colnames(taxa_count)[count_cols]
taxa_count_species <- data.frame(Species = as.character(species_uniq), taxa_count_species.mat)

# Some summary statistics
species_stats <- data.frame(Species = as.character(taxa_count_species$Species), 
	q25_count = as.numeric(apply(taxa_count_species.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(taxa_count_species.mat, 1, median)),
	mean_count = as.numeric(apply(taxa_count_species.mat, 1, mean)), stringsAsFactors = FALSE)
summary(species_stats$median_count)
summary(species_stats$mean_count)
species_stats <- species_stats[order(species_stats$median_count, decreasing = TRUE),]

datatable(species_stats)

write.table(species_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/taxa_count_species_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
species_common <- species_stats$Species[which(species_stats$median_count > 0)]  # 57
print("Number of common species (median_count>0")
length(species_common)
```

### Identify common taxa
- n=57 species passing filters
```{r}
count_cols <- grep("BBK", colnames(taxa_count_species))
taxa_count_species_common <- taxa_count_species[which(taxa_count_species$Species %in% species_common),]

# Visualise the distribution
# 31 is for QAND01 which had the lowest median
common_species_reads<-colSums(taxa_count_species_common [,-1])
common_species_reads <-data.frame(common_species_reads)
common_species_reads$MG_SampleID<-rownames(common_species_reads) # turn rownames into a column
plot_counts<-plot_counts %>% dplyr::left_join (common_species_reads, by="MG_SampleID")
```

### Check distribution
- how many reads there are in each sample per species (common) and in total per group

#### Distribution of total common species count per group

```{r}
# Common_assigned - density plot - 4 groups
ggplot(data=plot_counts, aes(x=common_species_reads, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# Common_assigned - density plot - NC vs SCZ
ggplot(data=plot_counts, aes(x=common_species_reads, group=scz, fill=scz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# Common_assigned - density plot - AAP vs CLZ
ggplot(data=subset(plot_counts, scz %in% "scz"), aes(x=common_species_reads, group=clz, fill=clz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()


```

#### Common species reads per sample

```{r}
tmp <- round(length(species_common)/5)
dist1 <- species_common[1:tmp]
dist2 <- species_common[(tmp+1):(2*tmp)]
dist3 <- species_common[((2*tmp)+1):(3*tmp)]
dist4 <- species_common[((3*tmp)+1):(4*tmp)]
dist5 <- species_common[((4*tmp)+1):length(species_common)]

taxa_count_species_common.long <- reshape2::melt(taxa_count_species_common, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_common.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)
mu5 <- mu %>% filter(Species %in% dist5)

taxa_count_species_common.long1 <- taxa_count_species_common.long %>% filter(Species %in% dist1)
taxa_count_species_common.long2 <- taxa_count_species_common.long %>% filter(Species %in% dist2)
taxa_count_species_common.long3 <- taxa_count_species_common.long %>% filter(Species %in% dist3)
taxa_count_species_common.long4 <- taxa_count_species_common.long %>% filter(Species %in% dist4)
taxa_count_species_common.long5 <- taxa_count_species_common.long %>% filter(Species %in% dist5)

ggplot(taxa_count_species_common.long1, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_common.long2, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_common.long3, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
 ggplot(taxa_count_species_common.long4, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_common.long5, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation
- need to transpose matrix (individuals as rows) before clr-transformation
```{r}

# Perform clr
taxa_count_species_common_clr <- taxa_count_species_common
tmp <- as.matrix(t(taxa_count_species_common_clr[,2:ncol(taxa_count_species_common_clr)]))
taxa_count_species_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_species_common_clr.tmp<-t(taxa_count_species_common_clr.tmp) # transpose it back
class(taxa_count_species_common_clr.tmp) <- "matrix"

taxa_count_species_common_clr <- data.frame(Species = taxa_count_species_common$Species, taxa_count_species_common_clr.tmp)

```

### Check distribution after clr transformation

- Restu said should be fine to use so long as there is some centrality

```{r}

tmp <- round(length(species_common)/5)
dist1 <- species_common[1:tmp]
dist2 <- species_common[(tmp+1):(2*tmp)]
dist3 <- species_common[((2*tmp)+1):(3*tmp)]
dist4 <- species_common[((3*tmp)+1):(4*tmp)]
dist5 <- species_common[((4*tmp)+1):length(species_common)]

taxa_count_species_common_clr.long <- reshape2::melt(taxa_count_species_common_clr, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_common_clr.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)
mu5 <- mu %>% filter(Species %in% dist5)

taxa_count_species_common_clr.long1 <- taxa_count_species_common_clr.long %>% filter(Species %in% dist1)
taxa_count_species_common_clr.long2 <- taxa_count_species_common_clr.long %>% filter(Species %in% dist2)
taxa_count_species_common_clr.long3 <- taxa_count_species_common_clr.long %>% filter(Species %in% dist3)
taxa_count_species_common_clr.long4 <- taxa_count_species_common_clr.long %>% filter(Species %in% dist4)
taxa_count_species_common_clr.long5 <- taxa_count_species_common_clr.long %>% filter(Species %in% dist5)

ggplot(taxa_count_species_common_clr.long1, aes(x = Read_count, color = Species)) +
  geom_density() +
  theme(legend.position="bottom")
ggplot(taxa_count_species_common_clr.long2, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(taxa_count_species_common_clr.long3, aes(x = Read_count, color = Species)) +
#  geom_density() + ylim(c(0,2)) +
#  theme(legend.position="bottom")
ggplot(taxa_count_species_common_clr.long3, aes(x = Read_count, color = Species)) +
  geom_density() +
  theme(legend.position="bottom")
# ggplot(taxa_count_species_common_clr.long4, aes(x = Read_count, color = Species)) +
#  geom_density() +
#  theme(legend.position="bottom")
ggplot(taxa_count_species_common_clr.long4, aes(x = Read_count, color = Species)) +
  geom_density() + ylim(c(0,2)) +
  theme(legend.position="bottom")
ggplot(taxa_count_species_common_clr.long5, aes(x = Read_count, color = Species)) +
  geom_density() + ylim(c(0,0.1)) +
  theme(legend.position="bottom")
ggplot(taxa_count_species_common_clr.long5, aes(x = Read_count, color = Species)) +
  geom_density() + ylim(c(0,0.1)) 

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(taxa_count_species_common_clr$Species)
taxa_count_species_common_clr_t <- as.data.frame(t(taxa_count_species_common_clr[,-(1)]))
colnames(taxa_count_species_common_clr_t) <- n
taxa_count_species_common_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_species_common_clr_t), taxa_count_species_common_clr_t)
taxa_count_species_common_clr_t <- taxa_count_species_common_clr_t[order(taxa_count_species_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_species_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_common_clr_t[,2:ncol(taxa_count_species_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Collapse to genus

### Take family OTUs

```{r}

taxa_count <- taxa_count_keep

taxonomy <- taxa_count$GTDB.taxonomy
genus <- unlist(lapply(strsplit(taxonomy, ";"), function(x) x[6]))
genus <- substring(genus, 5)

taxa_count$Taxon <- genus
colnames(taxa_count)[which(colnames(taxa_count) == "Taxon")] <- "Genus"
taxa_count[which(taxa_count$GTDB.taxonomy == "UNASSIGNED"), "Genus"] <- "UNASSIGNED"

```

### Collapse

```{r}

genus_uniq <- unique(genus)
count_cols <- grep("BBK", colnames(taxa_count))

taxa_count_genus.mat <- matrix(nrow = length(genus_uniq), ncol = length(count_cols))
for (i in 1:length(genus_uniq)) {
	taxa_count_genus.mat[i,] <- colSums(taxa_count[which(taxa_count$Genus == genus_uniq[i]),count_cols])
}
colnames(taxa_count_genus.mat) <- colnames(taxa_count)[count_cols]
taxa_count_genus <- data.frame(Genus = as.character(genus_uniq), taxa_count_genus.mat)

# Some summary statistics
genus_stats <- data.frame(Genus = as.character(taxa_count_genus$Genus), 
	q25_count = as.numeric(apply(taxa_count_genus.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(taxa_count_genus.mat, 1, median)),
	mean_count = as.numeric(apply(taxa_count_genus.mat, 1, mean)), stringsAsFactors = FALSE)
summary(genus_stats$median_count)
summary(genus_stats$mean_count)
genus_stats <- genus_stats[order(genus_stats$median_count, decreasing = TRUE),]

datatable(genus_stats)

write.table(genus_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/taxa_count_genus_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
genus_common <- genus_stats$Genus[which(genus_stats$median_count > 0)]

```

### Identify common taxa
- 51 common genus
```{r}

count_cols <- grep("BBK", colnames(taxa_count_genus))

taxa_count_genus_common <- taxa_count_genus[which(taxa_count_genus$Genus %in% genus_common),]

```

### Check distribution

```{r, echo=FALSE, message=FALSE, warning=FALSE}

tmp <- round(length(genus_common)/3)
dist1 <- genus_common[1:tmp]
dist2 <- genus_common[(tmp+1):(2*tmp)]
dist3 <- genus_common[((2*tmp)+1):length(genus_common)]

taxa_count_genus_common.long <- reshape2::melt(taxa_count_genus_common, id.vars=c("Genus"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_genus_common.long, "Genus", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Genus %in% dist1)
mu2 <- mu %>% filter(Genus %in% dist2)
mu3 <- mu %>% filter(Genus %in% dist3)

taxa_count_genus_common.long1 <- taxa_count_genus_common.long %>% filter(Genus %in% dist1)
taxa_count_genus_common.long2 <- taxa_count_genus_common.long %>% filter(Genus %in% dist2)
taxa_count_genus_common.long3 <- taxa_count_genus_common.long %>% filter(Genus %in% dist3)

ggplot(taxa_count_genus_common.long1, aes(x = Read_count, color = Genus)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_genus_common.long2, aes(x = Read_count, color = Genus)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_genus_common.long3, aes(x = Read_count, color = Genus)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation

```{r}

# Perform clr
taxa_count_genus_common_clr <- taxa_count_genus_common
tmp <- as.matrix(t(taxa_count_genus_common_clr[,2:ncol(taxa_count_genus_common_clr)]))
taxa_count_genus_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_genus_common_clr.tmp<-t(taxa_count_genus_common_clr.tmp)   #transpose it back
class(taxa_count_genus_common_clr.tmp) <- "matrix"

taxa_count_genus_common_clr <- data.frame(Genus = taxa_count_genus_common$Genus, taxa_count_genus_common_clr.tmp)

```

### Check distribution after clr transformation

- Restu said should be fine to use so long as there is some centrality

```{r, echo=FALSE, message=FALSE, warning=FALSE}

tmp <- round(length(genus_common)/3)
dist1 <- genus_common[1:tmp]
dist2 <- genus_common[(tmp+1):(2*tmp)]
dist3 <- genus_common[((2*tmp)+1):length(genus_common)]

taxa_count_genus_common_clr.long <- reshape2::melt(taxa_count_genus_common_clr, id.vars=c("Genus"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_genus_common.long, "Genus", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Genus %in% dist1)
mu2 <- mu %>% filter(Genus %in% dist2)
mu3 <- mu %>% filter(Genus %in% dist3)

taxa_count_genus_common_clr.long1 <- taxa_count_genus_common_clr.long %>% filter(Genus %in% dist1)
taxa_count_genus_common_clr.long2 <- taxa_count_genus_common_clr.long %>% filter(Genus %in% dist2)
taxa_count_genus_common_clr.long3 <- taxa_count_genus_common_clr.long %>% filter(Genus %in% dist3)

ggplot(taxa_count_genus_common_clr.long1, aes(x = Read_count, color = Genus)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_genus_common_clr.long2, aes(x = Read_count, color = Genus)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_genus_common_clr.long3, aes(x = Read_count, color = Genus)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(taxa_count_genus_common_clr$Genus)
taxa_count_genus_common_clr_t <- as.data.frame(t(taxa_count_genus_common_clr[,-(1)]))
colnames(taxa_count_genus_common_clr_t) <- n
taxa_count_genus_common_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_genus_common_clr_t), taxa_count_genus_common_clr_t)
taxa_count_genus_common_clr_t <- taxa_count_genus_common_clr_t[order(taxa_count_genus_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

taxa_count_genus_common_clr_t <- taxa_count_genus_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_genus_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_genus_common_clr_t[,2:ncol(taxa_count_genus_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_genus_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Collapse to families

### Take family OTUs

```{r}

taxa_count <- taxa_count_keep

taxonomy <- taxa_count$GTDB.taxonomy
family <- unlist(lapply(strsplit(taxonomy, ";"), function(x) x[5]))
family <- substring(family, 5)

taxa_count$Taxon <- family
colnames(taxa_count)[which(colnames(taxa_count) == "Taxon")] <- "Family"
taxa_count[which(taxa_count$GTDB.taxonomy == "UNASSIGNED"), "Family"] <- "UNASSIGNED"

```

### Collapse

```{r}

family_uniq <- unique(family)
count_cols <- grep("BBK", colnames(taxa_count))

taxa_count_family.mat <- matrix(nrow = length(family_uniq), ncol = length(count_cols))
for (i in 1:length(family_uniq)) {
	taxa_count_family.mat[i,] <- colSums(taxa_count[which(taxa_count$Family == family_uniq[i]),count_cols])
}
colnames(taxa_count_family.mat) <- colnames(taxa_count)[count_cols]
taxa_count_family <- data.frame(Family = as.character(family_uniq), taxa_count_family.mat)

# Some summary statistics
family_stats <- data.frame(Family = as.character(taxa_count_family$Family), 
	q25_count = as.numeric(apply(taxa_count_family.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(taxa_count_family.mat, 1, median)),
	mean_count = as.numeric(apply(taxa_count_family.mat, 1, mean)), stringsAsFactors = FALSE)
summary(family_stats$median_count)
summary(family_stats$mean_count)
family_stats <- family_stats[order(family_stats$median_count, decreasing = TRUE),]

datatable(family_stats)

write.table(family_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/taxa_count_family_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
family_common <- family_stats[which(family_stats$median_count > 0),"Family"]

```

### Identify common taxa
- 24 common family 
```{r}

count_cols <- grep("BBK", colnames(taxa_count_family))

taxa_count_family_common <- taxa_count_family[which(taxa_count_family$Family %in% family_common),]

```

### Check distribution

```{r, echo=FALSE, message=FALSE, warning=FALSE}

tmp <- round(length(family_common)/3)
dist1 <- family_common[1:tmp]
dist2 <- family_common[(tmp+1):(2*tmp)]
dist3 <- family_common[((2*tmp)+1):length(family_common)]

taxa_count_family_common.long <- reshape2::melt(taxa_count_family_common, id.vars=c("Family"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_family_common.long, "Family", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Family %in% dist1)
mu2 <- mu %>% filter(Family %in% dist2)
mu3 <- mu %>% filter(Family %in% dist3)

taxa_count_family_common.long1 <- taxa_count_family_common.long %>% filter(Family %in% dist1)
taxa_count_family_common.long2 <- taxa_count_family_common.long %>% filter(Family %in% dist2)
taxa_count_family_common.long3 <- taxa_count_family_common.long %>% filter(Family %in% dist3)

ggplot(taxa_count_family_common.long1, aes(x = Read_count, color = Family)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_family_common.long2, aes(x = Read_count, color = Family)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_family_common.long3, aes(x = Read_count, color = Family)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation

```{r}

# Perform clr
taxa_count_family_common_clr <- taxa_count_family_common
tmp <- as.matrix(t(taxa_count_family_common_clr[,2:ncol(taxa_count_family_common_clr)]))
taxa_count_family_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_family_common_clr.tmp<-t(taxa_count_family_common_clr.tmp)   #transpose it back
class(taxa_count_family_common_clr.tmp) <- "matrix"

taxa_count_family_common_clr <- data.frame(Family = taxa_count_family_common$Family, taxa_count_family_common_clr.tmp)

```

### Check distribution after clr transformation

- Restu said should be fine to use so long as there is some centrality

```{r, echo=FALSE, message=FALSE, warning=FALSE}

tmp <- round(length(family_common)/3)
dist1 <- family_common[1:tmp]
dist2 <- family_common[(tmp+1):(2*tmp)]
dist3 <- family_common[((2*tmp)+1):length(family_common)]

taxa_count_family_common_clr.long <- reshape2::melt(taxa_count_family_common_clr, id.vars=c("Family"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_family_common.long, "Family", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Family %in% dist1)
mu2 <- mu %>% filter(Family %in% dist2)
mu3 <- mu %>% filter(Family %in% dist3)

taxa_count_family_common_clr.long1 <- taxa_count_family_common_clr.long %>% filter(Family %in% dist1)
taxa_count_family_common_clr.long2 <- taxa_count_family_common_clr.long %>% filter(Family %in% dist2)
taxa_count_family_common_clr.long3 <- taxa_count_family_common_clr.long %>% filter(Family %in% dist3)

ggplot(taxa_count_family_common_clr.long1, aes(x = Read_count, color = Family)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_family_common_clr.long2, aes(x = Read_count, color = Family)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_family_common_clr.long3, aes(x = Read_count, color = Family)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(taxa_count_family_common_clr$Family)
taxa_count_family_common_clr_t <- as.data.frame(t(taxa_count_family_common_clr[,-(1)]))
colnames(taxa_count_family_common_clr_t) <- n
taxa_count_family_common_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_family_common_clr_t), taxa_count_family_common_clr_t)
taxa_count_family_common_clr_t <- taxa_count_family_common_clr_t[order(taxa_count_family_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

taxa_count_family_common_clr_t <- taxa_count_family_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_family_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_family_common_clr_t[,2:ncol(taxa_count_family_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_family_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take ~all species (317, filter <10)

```{r}

taxa_count <- taxa_count_keep

count_cols <- grep("BBK", colnames(taxa_count))

# Filter 1: based on features across number of samples (<10 non-zero samples)
zero_count <- unlist(lapply(1:nrow(taxa_count), function(x) length(which(taxa_count[x,] == 0))))
hist(zero_count)
zero_rm <- which(zero_count > ((ncol(taxa_count) - 2) - 10))
zero_rm_taxa <- taxa_count$Taxon[zero_rm]

taxa_count_rm0 <- taxa_count[-zero_rm,]

# Filter 2: based on reads across number of samples (<100 assigned reads in at least one sample)
maxcount_reads <- as.numeric(apply(taxa_count[,count_cols],1, max))
min(maxcount_reads) # 343 --> don't remove any species based on this filter
zero_reads_rm <- which(maxcount_reads < 100) # There are none, min is 343

taxa_count_species_rm0 <- taxa_count_rm0
```


```{r}
# Ploting the distribution of reads
species_rm0_317_reads<-colSums(taxa_count_species_rm0[-318,3:ncol(taxa_count_species_rm0)])
species_rm0_317_reads<-data.frame(species_rm0_317_reads)
colnames(species_rm0_317_reads)<-c("species_rm0_317_reads")
species_rm0_317_reads$MG_SampleID<-rownames(species_rm0_317_reads) # turn rownames into a column
plot_counts<-plot_counts %>% dplyr::left_join (species_rm0_317_reads, by="MG_SampleID")


# species_rm0_317_reads - density plot - 4 groups
ggplot(data=plot_counts, aes(x=species_rm0_317_reads, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# species_rm0_317_reads - density plot - NC vs SCZ
ggplot(data=plot_counts, aes(x=species_rm0_317_reads, group=scz, fill=scz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# species_rm0_317_reads - density plot - AAP vs CLZ
ggplot(data=subset(plot_counts, scz %in% "scz"), aes(x=species_rm0_317_reads, group=clz, fill=clz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()



# from CY
#maxcount_reads_sample <- as.numeric(sapply(taxa_count[,count_cols], max)) - this finds the bacteria with highest count per sample

#taxa_count_species_rm0.tmp <- data.frame(clr(as.matrix(taxa_count_rm0[,3:ncol(taxa_count_rm0)])))
#taxa_count_species_rm0 <- data.frame(Species = taxa_count_rm0$Taxon, taxa_count_species_rm0.tmp)



```

### clr transformation

```{r}

# Perform clr
taxa_count_species_rm0_clr <- taxa_count_species_rm0
tmp <- as.matrix(t(taxa_count_species_rm0_clr[,3:ncol(taxa_count_species_rm0_clr)]))
taxa_count_species_rm0_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_species_rm0_clr.tmp <- t(taxa_count_species_rm0_clr.tmp) #transpose it back
class(taxa_count_species_rm0_clr.tmp) <- "matrix"

taxa_count_species_rm0_clr <- data.frame(Species = taxa_count_species_rm0$Taxon, taxa_count_species_rm0_clr.tmp)

```

### Check distribution

- below is only an illustration of 40
- point being that the distributions do look funny --> not sure whether it violates model assumptions

```{r}

tmp <- round(length(taxa_count_species_rm0_clr$Species)/10)
dist1 <- taxa_count_species_rm0_clr$Species[1:tmp]
dist2 <- taxa_count_species_rm0_clr$Species[(tmp+1):(2*tmp)]
dist3 <- taxa_count_species_rm0_clr$Species[((2*tmp)+1):(3*tmp)]
dist4 <- taxa_count_species_rm0_clr$Species[((3*tmp)+1):(4*tmp)]

taxa_count_species_rm0_clr.long <- reshape2::melt(taxa_count_species_rm0_clr, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_rm0_clr.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)

taxa_count_species_rm0_clr.long1 <- taxa_count_species_rm0_clr.long %>% filter(Species %in% dist1)
taxa_count_species_rm0_clr.long2 <- taxa_count_species_rm0_clr.long %>% filter(Species %in% dist2)
taxa_count_species_rm0_clr.long3 <- taxa_count_species_rm0_clr.long %>% filter(Species %in% dist3)
taxa_count_species_rm0_clr.long4 <- taxa_count_species_rm0_clr.long %>% filter(Species %in% dist4)

ggplot(taxa_count_species_rm0_clr.long1, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_clr.long2, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_clr.long3, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_clr.long4, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(taxa_count_species_rm0_clr$Species)
taxa_count_species_rm0_clr_t <- as.data.frame(t(taxa_count_species_rm0_clr[,-(1)]))
colnames(taxa_count_species_rm0_clr_t) <- n
taxa_count_species_rm0_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_species_rm0_clr_t), taxa_count_species_rm0_clr_t)
taxa_count_species_rm0_clr_t <- taxa_count_species_rm0_clr_t[order(taxa_count_species_rm0_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

taxa_count_species_rm0_clr_t <- taxa_count_species_rm0_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_species_rm0_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_rm0_clr_t[,2:(ncol(taxa_count_species_rm0_clr_t)-1)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_rm0_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take "rare" (but not zero) samples (260, <10)

- ie. median < 0, but <10 non-zero samples
- n = 260 pass filters

### Take species OTUs

```{r}

taxa_count <- taxa_count_keep

species <- taxa_count$Taxon
species <- substring(species, 4)

taxa_count$Taxon <- species
colnames(taxa_count)[which(colnames(taxa_count) == "Taxon")] <- "Species"
taxa_count[which(taxa_count$GTDB.taxonomy == "UNASSIGNED"), "Species"] <- "UNASSIGNED"

```

### Remove (super) zero-inflated

```{r}

count_cols <- grep("BBK", colnames(taxa_count))

# Filter 1: based on features across number of samples (<10 non-zero samples)
zero_count <- unlist(lapply(1:nrow(taxa_count), function(x) length(which(taxa_count[x,] == 0))))
hist(zero_count)
zero_rm <- which(zero_count > ((ncol(taxa_count) - 2) - 10))
zero_rm_taxa <- taxa_count$Taxon[zero_rm]

# Filter 2: based on reads across number of samples (<100 assigned reads in at least one sample)
maxcount_reads <- as.numeric(apply(taxa_count[,count_cols],1, max))
min(maxcount_reads) # 343 --> don't remove any species based on this filter
zero_reads_rm <- which(maxcount_reads < 100)

taxa_count_species_rm0 <- taxa_count[-zero_rm,]

```

### Identify rare taxa 
- 260 pass filters
```{r}

species_stats <- fread("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/taxa_count_species_stats.txt")

species_rare <- species_stats$Species[which(species_stats$median_count == 0)]

count_cols <- grep("BBK", colnames(taxa_count_species_rm0))

taxa_count_species_rm0_rare <- taxa_count_species_rm0[which(taxa_count_species_rm0$Species %in% species_rare),]

# Adding rm0 rare species reads to plot density 
rm0_rare_species_reads<-colSums(taxa_count_species_rm0_rare [,-c(1,2)])
rm0_rare_species_reads <-data.frame(rm0_rare_species_reads)
rm0_rare_species_reads$MG_SampleID<-rownames(rm0_rare_species_reads) # turn rownames into a column
plot_counts<-plot_counts %>% dplyr::left_join (rm0_rare_species_reads, by="MG_SampleID")

# rm0_rare_species_reads - density plot - 4 groups
ggplot(data=plot_counts, aes(x=rm0_rare_species_reads, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4) +
    scale_fill_manual(values=c("#D81B60","#21DAC9", "#FFC107", "#3A9A8A"))

# rm0_rare_species_reads - density plot - NC vs SCZ
ggplot(data=plot_counts, aes(x=rm0_rare_species_reads, group=scz, fill=scz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# rm0_rare_species_reads - density plot - AAP vs CLZ
ggplot(data=subset(plot_counts, scz %in% "scz"), aes(x=rm0_rare_species_reads, group=clz, fill=clz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()


```

### Presence/absence (0/1)

- if count = 0

```{r}

taxa_count_species_rm0_rare_01 <- taxa_count_species_rm0_rare
taxa_count_species_rm0_rare_01.tmp <- apply(taxa_count_species_rm0_rare_01[,3:ncol(taxa_count_species_rm0_rare_01)], 2, function(x) ifelse(x != 0, 1, 0))
taxa_count_species_rm0_rare_01 <- data.frame(Species = taxa_count_species_rm0_rare$Species, taxa_count_species_rm0_rare_01.tmp)

```

### clr transformation

```{r}

# Perform clr
taxa_count_species_rm0_rare_clr <- taxa_count_species_rm0_rare
tmp <- as.matrix(t(taxa_count_species_rm0_rare_clr[,3:ncol(taxa_count_species_rm0_rare_clr)]))
taxa_count_species_rm0_rare_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_species_rm0_rare_clr.tmp<-t(taxa_count_species_rm0_rare_clr.tmp) # transpose it back
class(taxa_count_species_rm0_rare_clr.tmp) <- "matrix"

taxa_count_species_rm0_rare_clr <- data.frame(Species = taxa_count_species_rm0_rare_clr$Species, taxa_count_species_rm0_rare_clr.tmp)

```

### Check distribution

- below is only an illustration of 40
- point being that the distributions do look funny --> not sure whether it violates model assumptions

```{r}

tmp <- round(length(taxa_count_species_rm0_rare_clr$Species)/10)
dist1 <- taxa_count_species_rm0_rare_clr$Species[1:tmp]
dist2 <- taxa_count_species_rm0_rare_clr$Species[(tmp+1):(2*tmp)]
dist3 <- taxa_count_species_rm0_rare_clr$Species[((2*tmp)+1):(3*tmp)]
dist4 <- taxa_count_species_rm0_rare_clr$Species[((3*tmp)+1):(4*tmp)]

taxa_count_species_rm0_rare_clr.long <- reshape2::melt(taxa_count_species_rm0_rare_clr, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_rm0_rare_clr.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)

taxa_count_species_rm0_rare_clr.long1 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist1)
taxa_count_species_rm0_rare_clr.long2 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist2)
taxa_count_species_rm0_rare_clr.long3 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist3)
taxa_count_species_rm0_rare_clr.long4 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist4)

ggplot(taxa_count_species_rm0_rare_clr.long1, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_rare_clr.long2, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_rare_clr.long3, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_rare_clr.long4, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(taxa_count_species_rm0_rare_clr$Species)
taxa_count_species_rm0_rare_clr_t <- as.data.frame(t(taxa_count_species_rm0_rare_clr[,-(1)]))
colnames(taxa_count_species_rm0_rare_clr_t) <- n
taxa_count_species_rm0_rare_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_species_rm0_rare_clr_t), taxa_count_species_rm0_rare_clr_t)
taxa_count_species_rm0_rare_clr_t <- taxa_count_species_rm0_rare_clr_t[order(taxa_count_species_rm0_rare_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

```{r}

# Read counts, keep 0s
n <- as.character(taxa_count_species_rm0_rare_01$Species)
taxa_count_species_rm0_rare_01_t <- as.data.frame(t(taxa_count_species_rm0_rare_01[,-(1)]))
colnames(taxa_count_species_rm0_rare_01_t) <- n
taxa_count_species_rm0_rare_01_t <- data.frame(MG_SampleID = rownames(taxa_count_species_rm0_rare_01_t), taxa_count_species_rm0_rare_01_t)
taxa_count_species_rm0_rare_01_t <- taxa_count_species_rm0_rare_01_t[order(taxa_count_species_rm0_rare_01_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

taxa_count_species_rm0_rare_clr_t <- taxa_count_species_rm0_rare_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)
taxa_count_species_rm0_rare_01_t <- taxa_count_species_rm0_rare_01_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_species_rm0_rare_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_rm0_rare_clr_t[,2:ncol(taxa_count_species_rm0_rare_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_rm0_rare_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

```{r}

# 1. Matrix
identical(taxa_count_species_rm0_rare_01_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_rm0_rare_01_t[,2:ncol(taxa_count_species_rm0_rare_01_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_rm0_rare_01_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```




## Take ~all species (519, filter <5)

```{r}


# Filter 1: based on features across number of samples (<5 non-zero samples)
zero_count <- unlist(lapply(1:nrow(taxa_count), function(x) length(which(taxa_count[x,] == 0))))
hist(zero_count)
zero_rm_filter_5 <- which(zero_count > ((ncol(taxa_count) - 2) - 5))
zero_rm_taxa_filter_5 <- taxa_count$Taxon[zero_rm_filter_5]

taxa_count_rm0_filter_5 <- taxa_count[-zero_rm_filter_5,]


taxa_count_species_rm0_filter_5 <- taxa_count_rm0_filter_5
```


```{r}
# Ploting the distribution of reads
species_rm0_519_reads<-colSums(taxa_count_species_rm0_filter_5[-520,3:ncol(taxa_count_species_rm0_filter_5)])
species_rm0_519_reads<-data.frame(species_rm0_519_reads)
colnames(species_rm0_519_reads)<-c("species_rm0_519_reads")
species_rm0_519_reads$MG_SampleID<-rownames(species_rm0_519_reads) # turn rownames into a column
plot_counts<-plot_counts %>% dplyr::left_join (species_rm0_519_reads, by="MG_SampleID")


# species_rm0_519_reads - density plot - 4 groups
ggplot(data=plot_counts, aes(x=species_rm0_519_reads, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# species_rm0_519_reads - density plot - NC vs SCZ
ggplot(data=plot_counts, aes(x=species_rm0_519_reads, group=scz, fill=scz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# species_rm0_519_reads - density plot - AAP vs CLZ
ggplot(data=subset(plot_counts, scz %in% "scz"), aes(x=species_rm0_519_reads, group=clz, fill=clz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()



# from CY
#maxcount_reads_sample <- as.numeric(sapply(taxa_count[,count_cols], max)) - this finds the bacteria with highest count per sample

#taxa_count_species_rm0.tmp <- data.frame(clr(as.matrix(taxa_count_rm0[,3:ncol(taxa_count_rm0)])))
#taxa_count_species_rm0 <- data.frame(Species = taxa_count_rm0$Taxon, taxa_count_species_rm0.tmp)



```

### clr transformation

```{r}

# Perform clr
taxa_count_species_rm0_filter_5_clr <- taxa_count_species_rm0_filter_5
tmp <- as.matrix(t(taxa_count_species_rm0_filter_5_clr[,3:ncol(taxa_count_species_rm0_filter_5_clr)]))
taxa_count_species_rm0_filter_5_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_species_rm0_filter_5_clr.tmp <- t(taxa_count_species_rm0_filter_5_clr.tmp) #transpose it back
class(taxa_count_species_rm0_filter_5_clr.tmp) <- "matrix"

taxa_count_species_rm0_filter_5_clr <- data.frame(Species = taxa_count_species_rm0_filter_5$Species, taxa_count_species_rm0_filter_5_clr.tmp)

```

### Check distribution

- below is only an illustration of 40
- point being that the distributions do look funny --> not sure whether it violates model assumptions

```{r}

tmp <- round(length(taxa_count_species_rm0_filter_5_clr$Species)/10)
dist1 <- taxa_count_species_rm0_filter_5_clr$Species[1:tmp]
dist2 <- taxa_count_species_rm0_filter_5_clr$Species[(tmp+1):(2*tmp)]
dist3 <- taxa_count_species_rm0_filter_5_clr$Species[((2*tmp)+1):(3*tmp)]
dist4 <- taxa_count_species_rm0_filter_5_clr$Species[((3*tmp)+1):(4*tmp)]

taxa_count_species_rm0_filter_5_clr.long <- reshape2::melt(taxa_count_species_rm0_filter_5_clr, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_rm0_filter_5_clr.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)

taxa_count_species_rm0_filter_5_clr.long1 <- taxa_count_species_rm0_filter_5_clr.long %>% filter(Species %in% dist1)
taxa_count_species_rm0_filter_5_clr.long2 <- taxa_count_species_rm0_filter_5_clr.long %>% filter(Species %in% dist2)
taxa_count_species_rm0_filter_5_clr.long3 <- taxa_count_species_rm0_filter_5_clr.long %>% filter(Species %in% dist3)
taxa_count_species_rm0_filter_5_clr.long4 <- taxa_count_species_rm0_filter_5_clr.long %>% filter(Species %in% dist4)

ggplot(taxa_count_species_rm0_filter_5_clr.long1, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_filter_5_clr.long2, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_filter_5_clr.long3, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_filter_5_clr.long4, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(taxa_count_species_rm0_filter_5_clr$Species)
taxa_count_species_rm0_filter_5_clr_t <- as.data.frame(t(taxa_count_species_rm0_filter_5_clr[,-(1)]))
colnames(taxa_count_species_rm0_filter_5_clr_t) <- n
taxa_count_species_rm0_filter_5_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_species_rm0_filter_5_clr_t), taxa_count_species_rm0_filter_5_clr_t)
taxa_count_species_rm0_filter_5_clr_t <- taxa_count_species_rm0_filter_5_clr_t[order(taxa_count_species_rm0_filter_5_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

taxa_count_species_rm0_filter_5_clr_t <- taxa_count_species_rm0_filter_5_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_species_rm0_filter_5_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_rm0_filter_5_clr_t[,2:(ncol(taxa_count_species_rm0_filter_5_clr_t)-1)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_rm0_filter_5_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take "rare" (but not zero) samples (462, filter<5)

- ie. median < 0, but <5 non-zero samples
- n = 462 pass filters

### Identify rare taxa 
- 462 pass filters
```{r}

species_stats <- fread("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/taxa_count_species_stats.txt")

species_rare <- species_stats$Species[which(species_stats$median_count == 0)]

count_cols <- grep("BBK", colnames(taxa_count_rm0_filter_5))

taxa_count_species_rm0_filter_5_rare <- taxa_count_rm0_filter_5[which(taxa_count_rm0_filter_5$Species %in% species_rare),]

# Adding rm0 rare species reads to plot density 
rm0_filter_5_rare_species_reads<-colSums(taxa_count_species_rm0_filter_5_rare [,-c(1,2)])
rm0_filter_5_rare_species_reads <-data.frame(rm0_filter_5_rare_species_reads)
rm0_filter_5_rare_species_reads$MG_SampleID<-rownames(rm0_filter_5_rare_species_reads) # turn rownames into a column
plot_counts<-plot_counts %>% dplyr::left_join (rm0_filter_5_rare_species_reads, by="MG_SampleID")

# rm0_filter_5_rare_species_reads - density plot - 4 groups
ggplot(data=plot_counts, aes(x=rm0_filter_5_rare_species_reads, group=participant_group, fill=participant_group)) +
    geom_density(adjust=1.5, alpha=.4) +
    scale_fill_manual(values=c("#D81B60","#21DAC9", "#FFC107", "#3A9A8A"))

# rm0_filter_5_rare_species_reads - density plot - NC vs SCZ
ggplot(data=plot_counts, aes(x=rm0_filter_5_rare_species_reads, group=scz, fill=scz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()

# rm0_filter_5_rare_species_reads - density plot - AAP vs CLZ
ggplot(data=subset(plot_counts, scz %in% "scz"), aes(x=rm0_filter_5_rare_species_reads, group=clz, fill=clz)) +
    geom_density(adjust=1.5, alpha=.4) +
    theme_ipsum()


```

### Presence/absence (0/1)

- if count = 0

```{r}

taxa_count_species_rm0_filter_5_rare_01 <- taxa_count_species_rm0_filter_5_rare
taxa_count_species_rm0_filter_5_rare_01.tmp <- apply(taxa_count_species_rm0_filter_5_rare_01[,3:ncol(taxa_count_species_rm0_filter_5_rare_01)], 2, function(x) ifelse(x != 0, 1, 0))
taxa_count_species_rm0_filter_5_rare_01 <- data.frame(Species = taxa_count_species_rm0_filter_5_rare$Species, taxa_count_species_rm0_filter_5_rare_01.tmp)

```

### clr transformation

```{r}

# Perform clr
taxa_count_species_rm0_filter_5_rare_clr <- taxa_count_species_rm0_filter_5_rare
tmp <- as.matrix(t(taxa_count_species_rm0_filter_5_rare_clr[,3:ncol(taxa_count_species_rm0_filter_5_rare_clr)]))
taxa_count_species_rm0_filter_5_rare_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
taxa_count_species_rm0_filter_5_rare_clr.tmp<-t(taxa_count_species_rm0_filter_5_rare_clr.tmp)   #transpose it back
class(taxa_count_species_rm0_filter_5_rare_clr.tmp) <- "matrix"

taxa_count_species_rm0_filter_5_rare_clr <- data.frame(Species = taxa_count_species_rm0_filter_5_rare_clr$Species, taxa_count_species_rm0_filter_5_rare_clr.tmp)

```

### Check distribution

- below is only an illustration of 40
- point being that the distributions do look funny --> not sure whether it violates model assumptions

```{r}

tmp <- round(length(taxa_count_species_rm0_filter_5_rare_clr$Species)/10)
dist1 <- taxa_count_species_rm0_filter_5_rare_clr$Species[1:tmp]
dist2 <- taxa_count_species_rm0_filter_5_rare_clr$Species[(tmp+1):(2*tmp)]
dist3 <- taxa_count_species_rm0_filter_5_rare_clr$Species[((2*tmp)+1):(3*tmp)]
dist4 <- taxa_count_species_rm0_filter_5_rare_clr$Species[((3*tmp)+1):(4*tmp)]

taxa_count_species_rm0_filter_5_rare_clr.long <- reshape2::melt(taxa_count_species_rm0_filter_5_rare_clr, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_rm0_filter_5_rare_clr.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)

taxa_count_species_rm0_filter_5_rare_clr.long1 <- taxa_count_species_rm0_filter_5_rare_clr.long %>% filter(Species %in% dist1)
taxa_count_species_rm0_filter_5_rare_clr.long2 <- taxa_count_species_rm0_filter_5_rare_clr.long %>% filter(Species %in% dist2)
taxa_count_species_rm0_filter_5_rare_clr.long3 <- taxa_count_species_rm0_filter_5_rare_clr.long %>% filter(Species %in% dist3)
taxa_count_species_rm0_filter_5_rare_clr.long4 <- taxa_count_species_rm0_filter_5_rare_clr.long %>% filter(Species %in% dist4)

ggplot(taxa_count_species_rm0_filter_5_rare_clr.long1, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_filter_5_rare_clr.long2, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_filter_5_rare_clr.long3, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_filter_5_rare_clr.long4, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep0s,  clr dataframe
n <- as.character(taxa_count_species_rm0_filter_5_rare_clr$Species)
taxa_count_species_rm0_filter_5_rare_clr_t <- as.data.frame(t(taxa_count_species_rm0_filter_5_rare_clr[,-(1)]))
colnames(taxa_count_species_rm0_filter_5_rare_clr_t) <- n
taxa_count_species_rm0_filter_5_rare_clr_t <- data.frame(MG_SampleID = rownames(taxa_count_species_rm0_filter_5_rare_clr_t), taxa_count_species_rm0_filter_5_rare_clr_t)
taxa_count_species_rm0_filter_5_rare_clr_t <- taxa_count_species_rm0_filter_5_rare_clr_t[order(taxa_count_species_rm0_filter_5_rare_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

```{r}

# Read counts, keep 0s, non-clr
n <- as.character(taxa_count_species_rm0_filter_5_rare_01$Species)
taxa_count_species_rm0_filter_5_rare_01_t <- as.data.frame(t(taxa_count_species_rm0_filter_5_rare_01[,-(1)]))
colnames(taxa_count_species_rm0_filter_5_rare_01_t) <- n
taxa_count_species_rm0_filter_5_rare_01_t <- data.frame(MG_SampleID = rownames(taxa_count_species_rm0_filter_5_rare_01_t), taxa_count_species_rm0_filter_5_rare_01_t)
taxa_count_species_rm0_filter_5_rare_01_t <- taxa_count_species_rm0_filter_5_rare_01_t[order(taxa_count_species_rm0_filter_5_rare_01_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

taxa_count_species_rm0_filter_5_rare_clr_t <- taxa_count_species_rm0_filter_5_rare_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)
taxa_count_species_rm0_filter_5_rare_01_t <- taxa_count_species_rm0_filter_5_rare_01_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(taxa_count_species_rm0_filter_5_rare_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_rm0_filter_5_rare_clr_t[,2:ncol(taxa_count_species_rm0_filter_5_rare_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_rm0_filter_5_rare_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

```{r}

# 1. Matrix
identical(taxa_count_species_rm0_filter_5_rare_01_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, taxa_count_species_rm0_filter_5_rare_01_t[,2:ncol(taxa_count_species_rm0_filter_5_rare_01_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/taxa_count_species_rm0_filter_5_rare_01_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```


## Take common EC level 4 variables

- Enzyme commission naming system: 4 levels, with increasingly finer classification (eg. 1.1.1.1, if denoting a higher level --> 1.1.-.-)
- Level 4 classification: n=2248 --> n=1600 after filtering for median > 0
- Classification is based on the chemical reactions that the enzymes catalyses 
- https://en.wikipedia.org/wiki/Enzyme_Commission_number

### Subset to desired enzyme level

```{r}

ec <- ec_keep

keep_level <- grep("[1-9].[1-9].[1-9].[1-9]", ec$VariableID)
ec <- ec[keep_level,]

```

### Identify common ECs

```{r}

count_cols <- grep("BBK", colnames(ec))
ec.mat <- ec[,count_cols]

# Some summary statistics
ec_stats <- data.frame(VariableID = as.character(ec$VariableID), 
	q25_count = as.numeric(apply(ec.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(ec.mat, 1, median)),
	mean_count = as.numeric(apply(ec.mat, 1, mean)), stringsAsFactors = FALSE)
summary(ec_stats$median_count)
summary(ec_stats$mean_count)
ec_stats <- ec_stats[order(ec_stats$median_count, decreasing = TRUE),]

datatable(ec_stats)

write.table(ec_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_EClevel4_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
ecid_common <- ec_stats$VariableID[which(ec_stats$median_count > 0)]

```

```{r}

count_cols <- grep("BBK", colnames(ecid_common))

ec_common <- ec[which(ec$VariableID %in% ecid_common),]

```

### Check distribution

- show a subset of EC with high median count and a subset of EC with low median count
- observe that EC with higher median count --> more normal distribution

```{r}

tmp <- round(length(ecid_common)/40)
dist1 <- ecid_common[1:tmp]
dist2 <- ecid_common[(tmp+1):(2*tmp)]
dist3 <- ecid_common[((2*tmp)+1):(3*tmp)]
dist4 <- ecid_common[((3*tmp)+1):(4*tmp)]
dist5 <- ecid_common[((30*tmp)+1):(31*tmp)]
dist6 <- ecid_common[((31*tmp)+1):(32*tmp)]
dist7 <- ecid_common[((33*tmp)+1):(34*tmp)]
dist8 <- ecid_common[((34*tmp)+1):(35*tmp)]

ec_common.long <- reshape2::melt(ec_common, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

ec_common.long1 <- ec_common.long %>% filter(VariableID %in% dist1)
ec_common.long2 <- ec_common.long %>% filter(VariableID %in% dist2)
ec_common.long3 <- ec_common.long %>% filter(VariableID %in% dist3)
ec_common.long4 <- ec_common.long %>% filter(VariableID %in% dist4)
ec_common.long5 <- ec_common.long %>% filter(VariableID %in% dist5)
ec_common.long6 <- ec_common.long %>% filter(VariableID %in% dist6)
ec_common.long7 <- ec_common.long %>% filter(VariableID %in% dist7)
ec_common.long8 <- ec_common.long %>% filter(VariableID %in% dist8)

ggplot(ec_common.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common.long6, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common.long7, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common.long8, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
```

### clr transformation

```{r}

# Perform clr
ec_common_clr <- ec_common
tmp <- as.matrix(t(ec_common_clr[,2:ncol(ec_common_clr)]))
ec_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
ec_common_clr.tmp <- t(ec_common_clr.tmp) # transpose it back
class(ec_common_clr.tmp) <- "matrix"
ec_common_clr <- data.frame(VariableID = ec_common$VariableID, ec_common_clr.tmp)

```

### Check distribution

- show a subset of EC with high median count and a subset of EC with low median count
- observe that EC with higher median count --> more normal distribution

```{r}

tmp <- round(length(ecid_common)/40)
dist1 <- ecid_common[1:tmp]
dist2 <- ecid_common[(tmp+1):(2*tmp)]
dist3 <- ecid_common[((2*tmp)+1):(3*tmp)]
dist4 <- ecid_common[((3*tmp)+1):(4*tmp)]
dist5 <- ecid_common[((30*tmp)+1):(31*tmp)]
dist6 <- ecid_common[((31*tmp)+1):(32*tmp)]
dist7 <- ecid_common[((33*tmp)+1):(34*tmp)]
dist8 <- ecid_common[((34*tmp)+1):(35*tmp)]

ec_common_clr.long <- reshape2::melt(ec_common_clr, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

ec_common_clr.long1 <- ec_common_clr.long %>% filter(VariableID %in% dist1)
ec_common_clr.long2 <- ec_common_clr.long %>% filter(VariableID %in% dist2)
ec_common_clr.long3 <- ec_common_clr.long %>% filter(VariableID %in% dist3)
ec_common_clr.long4 <- ec_common_clr.long %>% filter(VariableID %in% dist4)
ec_common_clr.long5 <- ec_common_clr.long %>% filter(VariableID %in% dist5)
ec_common_clr.long6 <- ec_common_clr.long %>% filter(VariableID %in% dist6)
ec_common_clr.long7 <- ec_common_clr.long %>% filter(VariableID %in% dist7)
ec_common_clr.long8 <- ec_common_clr.long %>% filter(VariableID %in% dist8)

ggplot(ec_common_clr.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
#ggplot(ec_common_clr.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common_clr.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
#ggplot(ec_common_clr.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common_clr.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
#ggplot(ec_common_clr.long6, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common_clr.long7, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common_clr.long8, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(ec_common_clr$VariableID)
ec_common_clr_t <- as.data.frame(t(ec_common_clr[,-(1)]))
colnames(ec_common_clr_t) <- n
ec_common_clr_t <- data.frame(MG_SampleID = rownames(ec_common_clr_t), ec_common_clr_t)
ec_common_clr_t <- ec_common_clr_t[order(ec_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

ec_common_clr_t <- ec_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

common EC4 = 1600

```{r}

# 1. Matrix
identical(ec_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out_common_EC4 <- data.table(FID = diet$participant_id, IID = diet$participant_id, ec_common_clr_t[,2:ncol(ec_common_clr_t)])
write.table(osca_out_common_EC4, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_EClevel4_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take "rare" (but not zero) samples

- ie. median < 0, but <10 non-zero samples
- n = 355 pass filters

### Subset to desired enzyme level

```{r}

ec <- ec_keep

keep_level <- grep("[1-9].[1-9].[1-9].[1-9]", ec$VariableID)
ec <- ec[keep_level,]

```

### Remove (super) zero-inflated

```{r}

count_cols <- grep("BBK", colnames(ec))

# Filter 1: based on features across number of samples (<10 non-zero samples)
zero_count <- unlist(lapply(1:nrow(ec), function(x) length(which(ec[x,] == 0))))
hist(zero_count)
zero_rm <- which(zero_count > ((ncol(ec) - 2) - 10))
zero_rm_taxa <- ec$Taxon[zero_rm]

# Filter 2: based on reads across number of samples (<100 assigned reads in at least one sample)
maxcount_reads <- as.numeric(sapply(ec[,count_cols], max))
zero_reads_rm <- which(maxcount_reads < 100)

ec_rm0 <- ec[-zero_rm,]

```

### Identify rare EC 

```{r}

ec_stats <- fread("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_EClevel4_stats.txt")

ec_rare <- ec_stats$VariableID[which(ec_stats$median_count == 0)]

count_cols <- grep("BBK", colnames(ec_rm0))

ec_rm0_rare <- ec_rm0[which(ec_rm0$VariableID %in% ec_rare),]

```

### Presence/absence (0/1)

- if count = 0

```{r}

ec_rm0_rare_01 <- ec_rm0_rare
ec_rm0_rare_01.tmp <- apply(ec_rm0_rare_01[,2:ncol(ec_rm0_rare_01)], 2, function(x) ifelse(x != 0, 1, 0))
ec_rm0_rare_01 <- data.frame(VariableID = ec_rm0_rare$VariableID, ec_rm0_rare_01.tmp)

```

### clr transformation

```{r}

# Perform clr
ec_rm0_rare_clr <- ec_rm0_rare
tmp <- as.matrix(t(ec_rm0_rare_clr[,2:ncol(ec_rm0_rare_clr)]))
ec_rm0_rare_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
ec_rm0_rare_clr.tmp <- t(ec_rm0_rare_clr.tmp) # transpose it back
class(ec_rm0_rare_clr.tmp) <- "matrix"

ec_rm0_rare_clr <- data.frame(VariableID = ec_rm0_rare$VariableID, ec_rm0_rare_clr.tmp)

```

### Check distribution

```{r}

tmp <- round(length(taxa_count_species_rm0_rare_clr$Species)/10)
dist1 <- taxa_count_species_rm0_rare_clr$Species[1:tmp]
dist2 <- taxa_count_species_rm0_rare_clr$Species[(tmp+1):(2*tmp)]
dist3 <- taxa_count_species_rm0_rare_clr$Species[((2*tmp)+1):(3*tmp)]
dist4 <- taxa_count_species_rm0_rare_clr$Species[((3*tmp)+1):(4*tmp)]

taxa_count_species_rm0_rare_clr.long <- reshape2::melt(taxa_count_species_rm0_rare_clr, id.vars=c("Species"), variable.name="ID", value.name = "Read_count")
library(plyr)
mu <- ddply(taxa_count_species_rm0_rare_clr.long, "Species", summarise, grp.mean=mean(Read_count))
mu1 <- mu %>% filter(Species %in% dist1)
mu2 <- mu %>% filter(Species %in% dist2)
mu3 <- mu %>% filter(Species %in% dist3)
mu4 <- mu %>% filter(Species %in% dist4)

taxa_count_species_rm0_rare_clr.long1 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist1)
taxa_count_species_rm0_rare_clr.long2 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist2)
taxa_count_species_rm0_rare_clr.long3 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist3)
taxa_count_species_rm0_rare_clr.long4 <- taxa_count_species_rm0_rare_clr.long %>% filter(Species %in% dist4)

ggplot(taxa_count_species_rm0_rare_clr.long1, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(taxa_count_species_rm0_rare_clr.long2, aes(x = Read_count, color = Species)) +
#   geom_density() + 
#   theme(legend.position="bottom")
# ggplot(taxa_count_species_rm0_rare_clr.long3, aes(x = Read_count, color = Species)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(taxa_count_species_rm0_rare_clr.long4, aes(x = Read_count, color = Species)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(ec_rm0_rare_clr$VariableID)
ec_rm0_rare_clr_t <- as.data.frame(t(ec_rm0_rare_clr[,-(1)]))
colnames(ec_rm0_rare_clr_t) <- n
ec_rm0_rare_clr_t <- data.frame(MG_SampleID = rownames(ec_rm0_rare_clr_t), ec_rm0_rare_clr_t)
ec_rm0_rare_clr_t <- ec_rm0_rare_clr_t[order(ec_rm0_rare_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

```{r}

# Read counts, keep 0s
n <- as.character(ec_rm0_rare_01$VariableID)
ec_rm0_rare_01_t <- as.data.frame(t(ec_rm0_rare_01[,-(1)]))
colnames(ec_rm0_rare_01_t) <- n
ec_rm0_rare_01_t <- data.frame(MG_SampleID = rownames(ec_rm0_rare_01_t), ec_rm0_rare_01_t)
ec_rm0_rare_01_t <- ec_rm0_rare_01_t[order(ec_rm0_rare_01_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

ec_rm0_rare_clr_t <- ec_rm0_rare_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)
ec_rm0_rare_01_t <- ec_rm0_rare_01_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(ec_rm0_rare_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, ec_rm0_rare_clr_t[,2:ncol(ec_rm0_rare_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_EClevel4_rm0_rare_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

```{r}

# 1. Matrix
identical(ec_rm0_rare_01_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, ec_rm0_rare_01_t[,2:ncol(ec_rm0_rare_01_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_EClevel4_rm0_rare_01_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take common EC level 3 variables

- Enzyme commission naming system: 4 levels, with increasingly finer classification (eg. 1.1.1.1, if denoting a higher level --> 1.1.-.-)
- Level 3 classification: n=98 --> n=89 after filtering for median > 0
- Classification is based on the chemical reactions that the enzymes catalyses 
- https://en.wikipedia.org/wiki/Enzyme_Commission_number

### Subset to desired enzyme level

```{r}

ec <- ec_keep

keep_level <- grep("[1-9].[1-9].[1-9].-", ec$VariableID)
ec <- ec[keep_level,]

```

### Identify common ECs

```{r}

count_cols <- grep("BBK", colnames(ec))
ec.mat <- ec[,count_cols]

# Some summary statistics
ec_stats <- data.frame(VariableID = as.character(ec$VariableID), 
	q25_count = as.numeric(apply(ec.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(ec.mat, 1, median)),
	mean_count = as.numeric(apply(ec.mat, 1, mean)), stringsAsFactors = FALSE)
summary(ec_stats$median_count)
summary(ec_stats$mean_count)
ec_stats <- ec_stats[order(ec_stats$median_count, decreasing = TRUE),]

datatable(ec_stats)

write.table(ec_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_EClevel3_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
ecid_common <- ec_stats$VariableID[which(ec_stats$median_count > 0)]

```

```{r}

count_cols <- grep("BBK", colnames(ecid_common))

ec_common <- ec[which(ec$VariableID %in% ecid_common),]

```

### Check distribution

- show a subset of EC with high median count and a subset of EC with low median count
- observe that EC with higher median count --> more normal distribution

```{r}

tmp <- round(length(ecid_common)/5)
dist1 <- ecid_common[1:tmp]
dist2 <- ecid_common[(tmp+1):(2*tmp)]
dist3 <- ecid_common[((2*tmp)+1):(3*tmp)]
dist4 <- ecid_common[((3*tmp)+1):(4*tmp)]
dist5 <- ecid_common[((3*tmp)+1):(5*tmp)]

ec_common.long <- reshape2::melt(ec_common, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

ec_common.long1 <- ec_common.long %>% filter(VariableID %in% dist1)
ec_common.long2 <- ec_common.long %>% filter(VariableID %in% dist2)
ec_common.long3 <- ec_common.long %>% filter(VariableID %in% dist3)
ec_common.long4 <- ec_common.long %>% filter(VariableID %in% dist4)
ec_common.long5 <- ec_common.long %>% filter(VariableID %in% dist5)

ggplot(ec_common.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation

```{r}

# Perform clr
ec_common_clr <- ec_common
tmp <- as.matrix(t(ec_common_clr[,2:ncol(ec_common_clr)]))
ec_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
ec_common_clr.tmp <-t(ec_common_clr.tmp) # transpose it back
class(ec_common_clr.tmp) <- "matrix"
ec_common_clr <- data.frame(VariableID = ec_common_clr$VariableID, ec_common_clr.tmp)

```

### Check distribution

- show a subset of EC with high median count and a subset of EC with low median count
- observe that EC with higher median count --> more normal distribution

```{r}

tmp <- round(length(ecid_common)/5)
dist1 <- ecid_common[1:tmp]
dist2 <- ecid_common[(tmp+1):(2*tmp)]
dist3 <- ecid_common[((2*tmp)+1):(3*tmp)]
dist4 <- ecid_common[((3*tmp)+1):(4*tmp)]
dist5 <- ecid_common[((4*tmp)+1):(5*tmp)]

ec_common_clr.long <- reshape2::melt(ec_common_clr, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

ec_common_clr.long1 <- ec_common_clr.long %>% filter(VariableID %in% dist1)
ec_common_clr.long2 <- ec_common_clr.long %>% filter(VariableID %in% dist2)
ec_common_clr.long3 <- ec_common_clr.long %>% filter(VariableID %in% dist3)
ec_common_clr.long4 <- ec_common_clr.long %>% filter(VariableID %in% dist4)
ec_common_clr.long5 <- ec_common_clr.long %>% filter(VariableID %in% dist5)

ggplot(ec_common_clr.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common_clr.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common_clr.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(ec_common_clr.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(ec_common_clr.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(ec_common_clr$VariableID)
ec_common_clr_t <- as.data.frame(t(ec_common_clr[,-(1)]))
colnames(ec_common_clr_t) <- n
ec_common_clr_t <- data.frame(MG_SampleID = rownames(ec_common_clr_t), ec_common_clr_t)
ec_common_clr_t <- ec_common_clr_t[order(ec_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

ec_common_clr_t <- ec_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(ec_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, ec_common_clr_t[,2:ncol(ec_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_EClevel3_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take TCDB

- "The TC system consists of a set of representative protein sequences, most of which have been functionally characterized. These transporters are classified with a five-character designation, as follows: D1.L1.D2.D3.D4. D1 (a single digit) corresponds to the transporter class (i.e. channel, carrier, primary active transporter, group translocator or transmembrane electron flow carrier). L1 (a letter) corresponds to the transporter subclass, which, e.g. in the case of primary active transporters, refers to the energy source used to drive transport. D2 (a number) corresponds to the transporter family (sometimes actually a superfamily). D3 (a number) corresponds to the subfamily (or the family of a superfamily) in which a transporter is found. D4 (a number) corresponds to the transporter itself. This refers to a specific transport system with a defined range of substrates, a known polarity of transport, an energy source that drives vectorial movement of the substrate and a mechanism of action. Only in one of the TC classes (class 9) is this information incomplete or absent." https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1334385/
- All TCDB families in this dataset go to 5 levels of classes --> no need to subset as far as I can see
- goes from 1181 to 254

### Keep all (as all have 5 levels of hierarchy)

```{r}

tcdb <- tcdb_keep

```

### Identify common TCDB variables

```{r}

count_cols <- grep("BBK", colnames(tcdb))
tcdb.mat <- tcdb[,count_cols]

# Some summary statistics
tcdb_stats <- data.frame(VariableID = as.character(tcdb$VariableID), 
	q25_count = as.numeric(apply(tcdb.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(tcdb.mat, 1, median)),
	mean_count = as.numeric(apply(tcdb.mat, 1, mean)), stringsAsFactors = FALSE)
summary(tcdb_stats$median_count)
summary(tcdb_stats$mean_count)
tcdb_stats <- tcdb_stats[order(tcdb_stats$median_count, decreasing = TRUE),]

datatable(tcdb_stats)

write.table(tcdb_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_TCDB_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
tcdbid_common <- tcdb_stats$VariableID[which(tcdb_stats$median_count > 0)]

```

```{r}

count_cols <- grep("BBK", colnames(tcdb))

tcdb_common <- tcdb[which(tcdb$VariableID %in% tcdbid_common),]

```

### Check distribution

- show a subset of TCDB with high median count and a subset of EC with low median count
- observe that TCDB with higher median count --> more normal distribution

```{r}

tmp <- round(length(tcdbid_common)/30)
dist1 <- tcdbid_common[1:tmp]
dist2 <- tcdbid_common[(tmp+1):(2*tmp)]
dist3 <- tcdbid_common[((2*tmp)+1):(3*tmp)]
dist4 <- tcdbid_common[((3*tmp)+1):(4*tmp)]
dist5 <- tcdbid_common[((4*tmp)+1):(5*tmp)]

tcdb_common.long <- reshape2::melt(tcdb_common, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

tcdb_common.long1 <- tcdb_common.long %>% filter(VariableID %in% dist1)
tcdb_common.long2 <- tcdb_common.long %>% filter(VariableID %in% dist2)
tcdb_common.long3 <- tcdb_common.long %>% filter(VariableID %in% dist3)
tcdb_common.long4 <- tcdb_common.long %>% filter(VariableID %in% dist4)
tcdb_common.long5 <- tcdb_common.long %>% filter(VariableID %in% dist5)

ggplot(tcdb_common.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(tcdb_common.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(tcdb_common.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(tcdb_common.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(tcdb_common.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation

```{r}

# Perform clr
tcdb_common_clr <- tcdb_common
tmp <- as.matrix(t(tcdb_common_clr[,2:ncol(tcdb_common_clr)]))
tcdb_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
tcdb_common_clr.tmp <- t(tcdb_common_clr.tmp) #transpose it back
class(tcdb_common_clr.tmp) <- "matrix"

tcdb_common_clr <- data.frame(VariableID = tcdb_common_clr$VariableID, tcdb_common_clr.tmp)

```

### Check distribution

- show a subset of TCDB with high median count and a subset of TCDB with low median count


```{r}

tmp <- round(length(tcdbid_common)/30)
dist1 <- tcdbid_common[1:tmp]
dist2 <- tcdbid_common[(tmp+1):(2*tmp)]
dist3 <- tcdbid_common[((2*tmp)+1):(3*tmp)]
dist4 <- tcdbid_common[((3*tmp)+1):(4*tmp)]
dist5 <- tcdbid_common[((4*tmp)+1):(5*tmp)]

tcdb_common_clr.long <- reshape2::melt(tcdb_common_clr, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

tcdb_common_clr.long1 <- tcdb_common_clr.long %>% filter(VariableID %in% dist1)
tcdb_common_clr.long2 <- tcdb_common_clr.long %>% filter(VariableID %in% dist2)
tcdb_common_clr.long3 <- tcdb_common_clr.long %>% filter(VariableID %in% dist3)
tcdb_common_clr.long4 <- tcdb_common_clr.long %>% filter(VariableID %in% dist4)
tcdb_common_clr.long5 <- tcdb_common_clr.long %>% filter(VariableID %in% dist5)

ggplot(tcdb_common_clr.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(tcdb_common_clr.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(tcdb_common_clr.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(tcdb_common_clr.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(tcdb_common_clr.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(tcdb_common_clr$VariableID)
tcdb_common_clr_t <- as.data.frame(t(tcdb_common_clr[,-(1)]))
colnames(tcdb_common_clr_t) <- n
tcdb_common_clr_t <- data.frame(MG_SampleID = rownames(tcdb_common_clr_t), tcdb_common_clr_t)
tcdb_common_clr_t <- tcdb_common_clr_t[order(tcdb_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

tcdb_common_clr_t <- tcdb_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(tcdb_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, tcdb_common_clr_t[,2:ncol(tcdb_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_TCDB_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take MetaCyc Pathways

- start with 2870 pathways
- common pathways (median > 0): n = 578

### Keep all (as within pathway "units")

```{r}

mcpath <- mcpath_keep

```

### Identify common MetaCyc variables

```{r}

count_cols <- grep("BBK", colnames(mcpath))
mcpath.mat <- mcpath[,count_cols]

# Some summary statistics
mcpath_stats <- data.frame(VariableID = as.character(mcpath$VariableID), 
	q25_count = as.numeric(apply(mcpath.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(mcpath.mat, 1, median)),
	mean_count = as.numeric(apply(mcpath.mat, 1, mean)), stringsAsFactors = FALSE)
summary(mcpath_stats$median_count)
summary(mcpath_stats$mean_count)
mcpath_stats <- mcpath_stats[order(mcpath_stats$median_count, decreasing = TRUE),]

datatable(mcpath_stats)

write.table(mcpath_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_MetaCycpathway_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
mcpathid_common <- mcpath_stats$VariableID[which(mcpath_stats$median_count > 0)]

```

```{r}

count_cols <- grep("BBK", colnames(mcpath))

mcpath_common <- mcpath[which(mcpath$VariableID %in% mcpathid_common),]

```

### Check distribution

- show a subset of MetaCyc Pathways with high median count and a subset of EC with low median count
- observe that MetaCyc Pathways with higher median count --> more normal distribution

```{r}

tmp <- round(length(mcpathid_common)/30)
dist1 <- mcpathid_common[1:tmp]
dist2 <- mcpathid_common[(tmp+1):(2*tmp)]
dist3 <- mcpathid_common[((2*tmp)+1):(3*tmp)]
dist4 <- mcpathid_common[((3*tmp)+1):(4*tmp)]
dist5 <- mcpathid_common[((4*tmp)+1):(5*tmp)]

mcpath_common.long <- reshape2::melt(mcpath_common, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

mcpath_common.long1 <- mcpath_common.long %>% filter(VariableID %in% dist1)
mcpath_common.long2 <- mcpath_common.long %>% filter(VariableID %in% dist2)
mcpath_common.long3 <- mcpath_common.long %>% filter(VariableID %in% dist3)
mcpath_common.long4 <- mcpath_common.long %>% filter(VariableID %in% dist4)
mcpath_common.long5 <- mcpath_common.long %>% filter(VariableID %in% dist5)

ggplot(mcpath_common.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(mcpath_common.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(mcpath_common.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation

```{r}

# Perform clr
mcpath_common_clr <- mcpath_common
tmp <- as.matrix(t(mcpath_common_clr[,2:ncol(mcpath_common_clr)]))
mcpath_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
mcpath_common_clr.tmp <-t(mcpath_common_clr.tmp) #transpose it back
class(mcpath_common_clr.tmp) <- "matrix"

mcpath_common_clr <- data.frame(VariableID = mcpath_common_clr$VariableID, mcpath_common_clr.tmp)

```

### Check distribution

- show a subset of MCpath with high median count and a subset of MCpath with low median count
- relatively normal distribution
```{r}

tmp <- round(length(mcpathid_common)/30)
dist1 <- mcpathid_common[1:tmp]
dist2 <- mcpathid_common[(tmp+1):(2*tmp)]
dist3 <- mcpathid_common[((2*tmp)+1):(3*tmp)]
dist4 <- mcpathid_common[((3*tmp)+1):(4*tmp)]
dist5 <- mcpathid_common[((4*tmp)+1):(5*tmp)]

mcpath_common_clr.long <- reshape2::melt(mcpath_common_clr, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

mcpath_common_clr.long1 <- mcpath_common_clr.long %>% filter(VariableID %in% dist1)
mcpath_common_clr.long2 <- mcpath_common_clr.long %>% filter(VariableID %in% dist2)
mcpath_common_clr.long3 <- mcpath_common_clr.long %>% filter(VariableID %in% dist3)
mcpath_common_clr.long4 <- mcpath_common_clr.long %>% filter(VariableID %in% dist4)
mcpath_common_clr.long5 <- mcpath_common_clr.long %>% filter(VariableID %in% dist5)

ggplot(mcpath_common_clr.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common_clr.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(mcpath_common_clr.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common_clr.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(mcpath_common_clr.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r}

# Read counts, keep 0s non-clr dataframe
n <- as.character(mcpath_common_clr$VariableID)
mcpath_common_clr_t <- as.data.frame(t(mcpath_common_clr[,-(1)]))
colnames(mcpath_common_clr_t) <- n
mcpath_common_clr_t <- data.frame(MG_SampleID = rownames(mcpath_common_clr_t), mcpath_common_clr_t)
mcpath_common_clr_t <- mcpath_common_clr_t[order(mcpath_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r}

mcpath_common_clr_t <- mcpath_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r}

# 1. Matrix
identical(mcpath_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, mcpath_common_clr_t[,2:ncol(mcpath_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_MetaCycpathway_common_clrcheck_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take Microba common gene functional tables

- start with 4,250,332 genes
- common pathways (median > 0): n = 232,442

### Keep all (as within pathway "units")

```{r}

func <- func_keep

```

### Identify common functional variables

```{r eval=FALSE}

count_cols <- grep("BBK", colnames(func))
func.mat <- func[,count_cols]

# Some summary statistics
func_stats <- data.frame(VariableID = as.character(func$VariableID), 
	q25_count = as.numeric(apply(func.mat, 1, function(x) quantile(x, 0.25))),
	median_count = as.numeric(apply(func.mat, 1, median)),
	mean_count = as.numeric(apply(func.mat, 1, mean)), stringsAsFactors = FALSE)
summary(func_stats$median_count)
summary(func_stats$mean_count)
func_stats <- func_stats[order(func_stats$median_count, decreasing = TRUE),]

datatable(func_stats)

write.table(func_stats, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_Microba_stats.txt",
	col.names = T, row.names = F, quote = F, sep = "\t")

```

```{r eval=FALSE}

func_stats <- read.delim("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_Microba_stats.txt", header = T, as.is = T)

# Define "common" taxa as those with median > 0
# Have also generate 25th quartile column, 
# as this is more likely to be normally distributed 
# (or at least less zero-inflated)
funcid_common <- func_stats$VariableID[which(func_stats$median_count > 0)]

```

```{r eval=FALSE}

count_cols <- grep("BBK", colnames(func))

func_common <- func[which(func$VariableID %in% funcid_common),]

```

### Check distribution

- show a subset of microbial genes with high median count and a subset of EC with low median count
- observe that microbial genes with higher median count --> more normal distribution

```{r eval=FALSE}

tmp <- round(length(funcid_common)/30)
dist1 <- funcid_common[1:tmp]
dist2 <- funcid_common[(tmp+1):(2*tmp)]
dist3 <- funcid_common[((2*tmp)+1):(3*tmp)]
dist4 <- funcid_common[((3*tmp)+1):(4*tmp)]
dist5 <- funcid_common[((4*tmp)+1):(5*tmp)]

func_common.long <- reshape2::melt(func_common, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

func_common.long1 <- func_common.long %>% filter(VariableID %in% dist1)
func_common.long2 <- func_common.long %>% filter(VariableID %in% dist2)
func_common.long3 <- func_common.long %>% filter(VariableID %in% dist3)
func_common.long4 <- func_common.long %>% filter(VariableID %in% dist4)
func_common.long5 <- func_common.long %>% filter(VariableID %in% dist5)

ggplot(func_common.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(func_common.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(func_common.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### clr transformation

```{r eval=FALSE}

# Perform clr
func_common_clr <- func_common
tmp <- as.matrix(t(func_common_clr[,2:ncol(func_common_clr)]))
func_common_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
func_common_clr.tmp <- t(func_common_clr.tmp) #transpose it back
class(func_common_clr.tmp) <- "matrix"

func_common_clr <- data.frame(VariableID = func_common_clr$VariableID, func_common_clr.tmp)

```

### Check distribution

- show a subset of Microba genes with high median count and a subset with low median count
- observe that Microba genes with higher median count --> more normal distribution

```{r eval=FALSE}

tmp <- round(length(mcpathid_common)/30)
dist1 <- funcid_common[1:tmp]
dist2 <- funcid_common[(tmp+1):(2*tmp)]
dist3 <- funcid_common[((2*tmp)+1):(3*tmp)]
dist4 <- funcid_common[((3*tmp)+1):(4*tmp)]
dist5 <- funcid_common[((4*tmp)+1):(5*tmp)]

func_common_clr.long <- reshape2::melt(func_common_clr, id.vars=c("VariableID"), variable.name="ID", value.name = "Read_count")

func_common_clr.long1 <- func_common_clr.long %>% filter(VariableID %in% dist1)
func_common_clr.long2 <- func_common_clr.long %>% filter(VariableID %in% dist2)
func_common_clr.long3 <- func_common_clr.long %>% filter(VariableID %in% dist3)
func_common_clr.long4 <- func_common_clr.long %>% filter(VariableID %in% dist4)
func_common_clr.long5 <- func_common_clr.long %>% filter(VariableID %in% dist5)

ggplot(func_common_clr.long1, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common_clr.long2, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(func_common_clr.long3, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")
# ggplot(mcpath_common_clr.long4, aes(x = Read_count, color = VariableID)) +
#   geom_density() + 
#   theme(legend.position="bottom")
ggplot(func_common_clr.long5, aes(x = Read_count, color = VariableID)) +
  geom_density() + 
  theme(legend.position="bottom")

```

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r eval=FALSE}
# Read counts, keep 0s non-clr dataframe
n <- as.character(func_common_clr$VariableID)
func_common_clr_t <- as.data.frame(t(func_common_clr[,-(1)]))
colnames(func_common_clr_t) <- n
func_common_clr_t <- data.frame(MG_SampleID = rownames(func_common_clr_t), func_common_clr_t)
func_common_clr_t <- func_common_clr_t[order(func_common_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r eval=FALSE}

func_common_clr_t <- func_common_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r eval=FALSE}

# 1. Matrix
identical(func_common_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, func_common_clr_t[,2:ncol(func_common_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_Microba_common_clrcheck_t.tsv", col.names = T, row.names = F, sep = "\t", quote = F)

```

## Take Microba "rare" (but not zero) samples

- ie. median < 0, but <10 non-zero samples
- n = 1160311 pass filters

### Identify rare taxa 

```{r eval=FALSE}

func_stats <- fread("C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/func_count_Microba_stats.txt")

func_rare <- func_stats$VariableID[which(func_stats$median_count == 0)]

count_cols <- grep("BBK", colnames(func))

func_rare <- func[which(func$VariableID %in% func_rare),]

```


### Remove (super) zero-inflated

```{r eval=FALSE}

count_cols <- grep("BBK", colnames(func_rare))

# Filter 1: based on features across number of samples (<10 non-zero samples)
zero_count <- unlist(lapply(1:nrow(func_rare), function(x) length(which(func_rare[x,] == 0))))
hist(zero_count)
zero_rm <- which(zero_count > ((ncol(func_rare) - 2) - 10))
zero_rm_taxa <- func_rare$VariableID[zero_rm]

# Filter 2: based on reads across number of samples (<100 assigned reads in at least one sample)
# - this filter is not relevant to functional data
# maxcount_reads <- as.numeric(sapply(func_rare[,count_cols], max))
# zero_reads_rm <- which(maxcount_reads < 100)

func_rm0_rare <- func_rare[-zero_rm,]

```

### Presence/absence (0/1)

- if count = 0

```{r eval=FALSE}

func_rm0_rare_01 <- func_rm0_rare
func_rm0_rare_01.tmp <- apply(func_rm0_rare_01[,2:ncol(func_rm0_rare_01)], 2, function(x) ifelse(x != 0, 1, 0))
func_rm0_rare_01 <- data.frame(VariableID = func_rm0_rare$VariableID, func_rm0_rare_01.tmp)

```

### clr transformation

```{r eval=FALSE}

# Perform clr
func_rm0_rare_clr <- func_rm0_rare
tmp <- as.matrix(t(func_rm0_rare_clr[,2:ncol(func_rm0_rare_clr)]))
func_rm0_rare_clr.tmp <- as.matrix(logratio.transfo(tmp, logratio = "CLR", offset = 0.001))
func_rm0_rare_clr.tmp <- t(func_rm0_rare_clr.tmp) #transpose it back
class(func_rm0_rare_clr.tmp) <- "matrix"

func_rm0_rare_clr <- data.frame(VariableID = func_rm0_rare_clr$VariableID, func_rm0_rare_clr.tmp)

```

### Check distribution

- left out to reduce memory/time requirements

### Pre-OSCA data cleaning

#### Transpose the matrix for OSCA (input: taxa by columns)

```{r eval=FALSE}

# Read counts, keep 0s non-clr dataframe
n <- as.character(func_rm0_rare_clr$VariableID)
func_rm0_rare_clr_t <- as.data.frame(t(func_rm0_rare_clr[,-(1)]))
colnames(func_rm0_rare_clr_t) <- n
func_rm0_rare_clr_t <- data.frame(MG_SampleID = rownames(func_rm0_rare_clr_t), func_rm0_rare_clr_t)
func_rm0_rare_clr_t <- func_rm0_rare_clr_t[order(func_rm0_rare_clr_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

```{r eval=FALSE}

# Read counts, keep 0s
n <- as.character(func_rm0_rare_01$VariableID)
func_rm0_rare_01_t <- as.data.frame(t(func_rm0_rare_01[,-(1)]))
colnames(func_rm0_rare_01_t) <- n
func_rm0_rare_01_t <- data.frame(MG_SampleID = rownames(func_rm0_rare_01_t), func_rm0_rare_01_t)
func_rm0_rare_01_t <- func_rm0_rare_01_t[order(func_rm0_rare_01_t$MG_SampleID),] # sort to be sure to avoid mix-ups

```

#### Subset to individuals with dietary data

```{r eval=FALSE}

func_rm0_rare_clr_t <- func_rm0_rare_clr_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

```{r eval=FALSE}

func_rm0_rare_01_t <- func_rm0_rare_01_t %>% filter(MG_SampleID %in% diet$MG_SampleID)

```

### Generate microbiome matrix for OSCA

```{r eval=FALSE}

# 1. Matrix
identical(func_rm0_rare_clr_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, func_rm0_rare_clr_t[,2:ncol(func_rm0_rare_clr_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_Microba_rm0_rare_clrcheck_t.tsv", col.names = T, row.names = F, sep = "\t", quote = F)

```

```{r eval=FALSE}

# 1. Matrix
identical(func_rm0_rare_01_t$MG_SampleID, diet$MG_SampleID)
osca_out <- data.table(FID = diet$participant_id, IID = diet$participant_id, func_rm0_rare_01_t[,2:ncol(func_rm0_rare_01_t)])
write.table(osca_out, "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/files/func_count_Microba_rm0_rare_01_t.tsv", 	col.names = T, row.names = F, sep = "\t", quote = F)

```

## Generate pheno and covar outputs for OSCA

### 1. NC-SCZ analysis: pheno + covariate files
- IPAQ: 1 - Low, 2- Moderate, 3-High
- Stool: 1- constipation, 2 - normal, 3 -lacking fibre, 4-diarrhea
- scz/clz: 0-no, 1-yes
- pheno: NC/SCZ
- cov: sex; create both with and without bristol stool chart
- qcov: create both with and without dietary PCs

```{r}
#Re-order levels
diet$participant_group <- factor(diet$participant_group, levels=c('NC', 'AAP', 'CR', 'CNR'))
diet$ipaq_pacat <- factor(diet$ipaq_pacat, levels=c('Low', 'Moderate', 'High'))
#diet$stool_type_regroup <- factor(diet$stool_type_regroup, levels=c('constipation', 'normal', 'lacking fibre', 'diarrhea'))

# Make ipaq and stool variables with characters
# Stool type
#diet <- diet %>%
# mutate(stool_type_regroup = recode(stool_type_regroup, "constipation" = '1', "normal" = '2', "lacking fibre" = '3', "diarrhea" = '4' ))
diet$stool_type_regroup <- as.numeric(diet$stool_type_regroup)
# IPAQ
#diet <- diet %>%
#  dplyr::recode(ipaq_pacat, "Low" = '1', "Moderate" = '2', "High" = '3')
#diet$ipaq_pacat <- as.numeric(diet$ipaq_pacat)
diet$FID<-diet$participant_id
diet$IID<-diet$participant_id

pheno <- diet[,c("participant_id", "participant_id", "participant_group","PC1", "PC2","PC3", "PC4", "scz", "clz", "stool_type_regroup", "bmi", "ipaq_pacat", "sex", "ipaq_met", "panss_total", "arfs", "laxatives", "PPI", "metformin", "TAP", "mets_5_notes", "mets_2","sagis_10","sagis_constipation_only", "Total_BNF", "Total_AMH", "CLZ_factor_AMH", "cloz_level", "norcloz_level", "cloz_norcloz_ratio", "panss_total_int", "Total_BNF_int", "Total_AMH_int",  "CLZ_factor_AMH_int", "cloz_level_int", "norcloz_level_int", "cloz_norcloz_ratio_int", "acetate_clean_int", "butyrate_clean_int", "propionate_clean_int", "arfs_int", "cloz_dose", "cloz_cd_ratio", "cloz_dose_int", "cloz_cd_ratio_int", "duration_first_treatment", "duration_illness")]
pheno$scz <- ifelse(pheno$scz == "scz", 1, 0)
pheno$clz <- ifelse(pheno$clz == "clz", 1, 0)
pheno$participant_group<-as.numeric(pheno$participant_group)
pheno$sex <- ifelse(pheno$sex == "male", 1, 0)
colnames(pheno)[1:2] <- c("FID", "IID")


cov <- diet[,c("participant_id", "participant_id", "sex", "clz", "scz", "metformin", "laxatives", "PPI", "sagis_constipation_only", "mets_5_notes", "TAP", "participant_group")]
cov$metformin<-as.character(cov$metformin)
cov$laxatives<-as.character(cov$laxatives)
cov$PPI<-as.character(cov$PPI)
cov$sagis_constipation_only<-as.character(cov$sagis_constipation_only)
cov$mets_5_notes<-as.character(cov$mets_5_notes)
cov$TAP<-as.character(cov$TAP)
cov$participant_group<-as.character(cov$participant_group)
colnames(cov)[1:2] <- c("FID", "IID")

qcov <- diet[,c("participant_id", "participant_id", "age","bmi", "PC1", "PC2", "PC3", "PC4","stool_type_regroup", "ipaq_pacat")]
colnames(qcov)[1:2] <- c("FID", "IID")

# Pheno adjusted for sex + age + PC1 + PC2 + PC3 + PC4 (logistic regression)
pheno_resid <- glm(pheno$scz ~ sex + age + bmi + PC1 + PC2 + PC3 + PC4 , data = diet, family = "binomial")
summary(pheno_resid)
pheno$scz_adjsexagebmidiet <- residuals(pheno_resid)
class(pheno$participant_group)
```

```{r}

pheno$participant_group<-as.character(pheno$participant_group)
#pheno$clz<-as.character(pheno$clz)
#pheno$scz<-as.character(pheno$scz)

# Raw pheno
write.table(pheno %>% dplyr::select(c("FID", "IID", "participant_group")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_participant_group.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "scz")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_nc_scz.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::filter(participant_group %in% c("1","3","4")) %>% dplyr::select(c("FID", "IID", "scz")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_nc_clz.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "scz_adjsexagebmidiet")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_nc_scz_resid_sex_age_bmi_dietPC.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "clz")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_aap_clz.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "bmi")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_bmi.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "bmi_cat")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_bmi_cat.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(diet %>% dplyr::select(c("FID", "IID", "bmi_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_bmi_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "ipaq_pacat")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_ipaq_pacat.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "ipaq_met")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_ipaq_met.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(qcov %>% dplyr::select(c("FID", "IID", "age")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_age.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "sex")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "arfs")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_arfs.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "arfs_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_arfs_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
  # Metabolic Syndrome
write.table(pheno %>% dplyr::select(c("FID", "IID", "mets_5_notes")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_mets_5_notes.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "mets_2")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_mets_2.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
  # Sagis_10 (constipation only) 
write.table(pheno %>% dplyr::select(c("FID", "IID", "sagis_10")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sagis_10.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "sagis_constipation_only")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sagis_constipation_only.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# in SCZ only
write.table(pheno %>% dplyr::select(c("FID", "IID", "panss_total")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_panss_total.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "Total_AMH")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_amh.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "Total_BNF")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_bnf.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "duration_first_treatment")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_duration_first_treatment.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "duration_illness")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_duration_illness.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# in SCZ only: int
write.table(pheno %>% dplyr::select(c("FID", "IID", "panss_total_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_panss_total_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "Total_AMH_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_amh_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "Total_BNF_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_bnf_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# In CLZ only
write.table(pheno %>% dplyr::select(c("FID", "IID", "CLZ_factor_AMH")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_clz_factor_AMH.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_level")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_level.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "norcloz_level")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_norcloz_level.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_norcloz_ratio")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_norcloz_ratio.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_dose")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_dose.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_cd_ratio")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_cd_ratio.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)


# In CLZ only: int
write.table(pheno %>% dplyr::select(c("FID", "IID", "CLZ_factor_AMH_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_clz_factor_amh_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_level_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_level_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "norcloz_level_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_norcloz_level_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_norcloz_ratio_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_norcloz_ratio_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_dose_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_dose_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "cloz_cd_ratio_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_cloz_cd_ratio_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

```


```{r}

# Discrete covariates (cov)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex", "metformin")),
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_metformin.cov",
	col.names = F, row.names = F, sep = "\t", quote = F) # sex, metformin, 
write.table(cov %>% dplyr::select(c("FID", "IID", "sex", "sagis_constipation_only")),
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_constipation.cov",
	col.names = F, row.names = F, sep = "\t", quote = F) # sex, metformin, 
write.table(cov %>% dplyr::select(c("FID", "IID", "sex", "clz")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_clz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex", "metformin", "clz")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_metformin_clz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "scz")),
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_scz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "clz")),
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_clz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "scz", "sex")),
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_scz_sex.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "scz", "clz", "sex", "metformin")),
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_scz_clz_sex_metformin.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID","participant_group")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_participant_group.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID","mets_5_notes")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_mets_5_notes.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","mets_5_notes")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_mets_5_notes.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","participant_group")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_participant_group.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","participant_group", "laxatives")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_participant_group_laxatives.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "laxatives")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_laxatives.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","participant_group", "sagis_constipation_only")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_participant_group_constipation.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sagis_constipation_only")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_constipation.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","mets_5_notes","participant_group")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_mets_5_notes_participant_group.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","mets_5_notes","clz")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_mets_5_notes_clz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID","mets_5_notes","clz")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_mets_5_notes_clz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID","sex","metformin")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_metformin.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","metformin","PPI","laxatives")),
  "C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_metformin_ppi_laxatives.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "sagis_constipation_only")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sagis_constipation_only.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID","sagis_constipation_only")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_sagis_constipation_only.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)



# Continuous covariates (qcov)
#write.table(qcov, 
#	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_age_dietPC_bmi_bristol_ipaq.qcov",
#	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(qcov %>% dplyr::select(-c("ipaq_pacat")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_age_dietPC_bmi_bristol.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(qcov %>% dplyr::select(-c("ipaq_pacat", "stool_type_regroup")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_age_dietPC_bmi.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(qcov %>% dplyr::select(-c("ipaq_pacat", "stool_type_regroup", "bmi")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_age_dietPC.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(qcov %>% dplyr::select(-c("ipaq_pacat", "PC1", "PC2", "PC3", "PC4", "stool_type_regroup")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_age.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "sagis_10")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sagis_10.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno %>% dplyr::select(c("FID", "IID", "panss_total")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_panss_total.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
# Total_BNF
write.table(pheno %>% dplyr::select(c("FID", "IID", "Total_BNF")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_bnf.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)
# Total_AMH
write.table(pheno %>% dplyr::select(c("FID", "IID", "Total_AMH")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_amh.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)

write.table(pheno %>% dplyr::select(c("FID", "IID", "PC1", "PC2", "PC3", "PC4", "bmi")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC_bmi.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)

write.table(pheno %>% dplyr::select(c("FID", "IID", "PC1", "bmi")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/bmi.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)

```

#### 1.1 Subsets
```{r}
# SCZ only
scz_only<-subset(pheno, scz=="1")
scz_only_list<-scz_only[ ,c(1,2)]
write.table(scz_only_list, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_subset_scz_only.list",
	col.names = F, row.names = F, sep = "\t", quote = F)

# CLZ only
clz_only<-pheno %>% filter(clz=="1")
clz_only_list<-clz_only[ ,c(1,2)]
write.table(clz_only_list, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_subset_clz_only.list",
	col.names = F, row.names = F, sep = "\t", quote = F)

# CLZ only with cloz_level data (46)
clz_only_46<-pheno %>% filter(cloz_level!="NA")
clz_only_46_list<-clz_only_46[ ,c(1,2)]
write.table(clz_only_46_list, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_subset_clz_only_46.list",
	col.names = F, row.names = F, sep = "\t", quote = F)

# No AAP only
no_aap_only<-subset(pheno, participant_group %in% c("1","3","4"))
no_aap_only_list<-no_aap_only[ ,c(1,2)]
write.table(no_aap_only_list, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_subset_no_aap_only.list",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Only individuals with SCZ who have metabolic syndrome
mets_yes<-subset(scz_only, mets_5_notes == "1")
mets_yes_list<-mets_yes[ ,c(1,2)]
write.table(mets_yes_list, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_subset_scz_mets_yes.list",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Only individuals with SCZ who have metabolic syndrome
constipation_no<-subset(scz_only, sagis_constipation_only == "0")
constipation_no_list<-constipation_no[ ,c(1,2)]
write.table(constipation_no_list, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_subset_scz_constipation_no.list",
	col.names = F, row.names = F, sep = "\t", quote = F)

```

### 2. Diet analysis (take PCs as the phenotype): pheno + covariate files

```{r}

pheno_diet <- diet[,c("FID", "IID", "PC1", "PC2", "PC3", "PC4", "PC1_int", "PC2_int", "PC3_int", "PC4_int")]
cov <- diet[,c("participant_id", "participant_id", "sex", "scz")]
colnames(cov)[1:2] <- c("FID", "IID")
# PC1 - regress out covariates
pheno_resid <- lm(pheno_diet$PC1 ~ cov$sex + cov$scz + qcov$age)
summary(pheno_resid)
pheno_diet$PC1_adjsexagegroup <- residuals(pheno_resid)
# PC2 - regress out covariates
pheno_resid <- lm(pheno_diet$PC2 ~ cov$sex + cov$scz + qcov$age)
summary(pheno_resid)
pheno_diet$PC2_adjsexagegroup <- residuals(pheno_resid)
# PC3 - regress out covariates
pheno_resid <- lm(pheno_diet$PC3 ~ cov$sex + cov$scz + qcov$age)
summary(pheno_resid)
pheno_diet$PC3_adjsexagegroup <- residuals(pheno_resid)
# PC4 - regress out covariates
pheno_resid <- lm(pheno_diet$PC4 ~ cov$sex + cov$scz + qcov$age)
summary(pheno_resid)
pheno_diet$PC4_adjsexagegroup <- residuals(pheno_resid)

# Pheno - raw
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC1")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC1.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC2")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC2.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC3")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC3.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC4")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC4.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Pheno - int
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC1_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC1_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC2_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC2_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC3_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC3_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC4_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC4_int.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Pheno - adjusted
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC1_adjsexagegroup")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC1_preadj.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC2_adjsexagegroup")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC2_preadj.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC3_adjsexagegroup")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC3_preadj.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_diet %>% dplyr::select(c("FID", "IID", "PC4_adjsexagegroup")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_dietPC4_preadj.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Discrete covariates
write.table(cov %>% dplyr::select(c("FID", "IID", "sex","scz")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_scz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)

```

### 3. Bristol Stool Chart analysis

```{r}

bristol <- diet %>% filter(!(is.na(stool_type_regroup)))
pheno_bss <- bristol[,c("participant_id", "participant_id", "stool_type_regroup")]
colnames(pheno_bss)[1:2] <- c("FID", "IID")
cov <- bristol[,c("participant_id", "participant_id", "sex", "scz")]
qcov <- bristol[,c("participant_id", "participant_id", "age", "bmi", "PC1", "PC2", "PC3", "PC4")]
colnames(cov)[1:2] <- c("FID", "IID")
colnames(qcov)[1:2] <- c("FID", "IID")

# Pheno
write.table(pheno_bss, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_bristol_6.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Discrete covariates
write.table(cov, 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_sex_scz.cov",
	col.names = F, row.names = F, sep = "\t", quote = F)

```

### 4. SCFAs as phenotype analysis

```{r}
pheno_scfa <- diet[,c("participant_id", "participant_id", "acetate_clean", "butyrate_clean", "propionate_clean")]
colnames(pheno_scfa)[1:2] <- c("FID", "IID")
cov <- diet[,c("participant_id", "participant_id", "sex", "scz")]
colnames(cov)[1:2] <- c("FID", "IID")

# Pheno 
write.table(pheno_scfa %>% dplyr::select(c("FID", "IID", "acetate_clean_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_acetate.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_scfa %>% dplyr::select(c("FID", "IID", "butyrate_clean_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_butyrate.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
write.table(pheno_scfa %>% dplyr::select(c("FID", "IID", "propionate_clean_int")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_propionate.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
```

### 5. Other phenos
```{r}

# Laxatives
write.table(pheno %>% dplyr::select(c("FID", "IID", "laxatives")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_laxatives.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
# Metformin
write.table(pheno %>% dplyr::select(c("FID", "IID", "metformin")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_metformin.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
# TAP
write.table(pheno %>% dplyr::select(c("FID", "IID", "TAP")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_tap.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)
# PPI
write.table(pheno %>% dplyr::select(c("FID", "IID", "PPI")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_ppi.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

# Reads
asreads_df <- plot_counts[,c("MG_SampleID","total_assigned", "common_species_reads", "rm0_rare_species_reads")] %>%
  left_join(diet[,c("MG_SampleID", "FID", "IID")], by="MG_SampleID") 

write.table(asreads_df %>% dplyr::select(c("FID", "IID", "total_assigned")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_assigned_reads.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

write.table(asreads_df %>% dplyr::select(c("FID", "IID", "total_assigned")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_total_assigned_reads.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)


write.table(asreads_df %>% dplyr::select(c("FID", "IID", "common_species_reads")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_common_species_reads.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

write.table(asreads_df %>% dplyr::select(c("FID", "IID", "common_species_reads")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_common_species_reads.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)

write.table(asreads_df %>% dplyr::select(c("FID", "IID", "rm0_rare_species_reads")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_rm0_rare_species_reads.pheno",
	col.names = F, row.names = F, sep = "\t", quote = F)

write.table(asreads_df %>% dplyr::select(c("FID", "IID", "rm0_rare_species_reads")), 
	"C:/Users/uqsvasil/Documents/PhD/TRS-BIOME/Data/OSCA/prep/osca_metagenomics_rm0_rare_species_reads.qcov",
	col.names = F, row.names = F, sep = "\t", quote = F)


```

*Save progress*

```{r}
save.image("osca_prep_30.06.22.Rdata")
load("osca_prep_30.06.22.Rdata")

```
